From dd03c69aaf83bfd5065e8c982c09f23a97f680ef Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 14 Jul 2012 15:15:42 +0800
Subject: [PATCH 1/8] .gitignore

---
 .gitignore |    1 +
 1 个文件被修改，插入 1 行(+)

diff --git a/.gitignore b/.gitignore
index 6fed1c6..f1ba9d8 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,6 @@
 /vlc/
 /android-libs/*.so
+/patches/100*.patch
 local.properties
 .DS_Store
 *.keystore
-- 
1.7.10.4

