From 9874e15819a8e06a552847b42214a02f2e46a206 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 11 Aug 2012 14:27:45 +0800
Subject: [PATCH 02/15] compile.sh: set CHOST for cross compile

---
 compile.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/compile.sh b/compile.sh
index 86cf38f..6247a75 100755
--- a/compile.sh
+++ b/compile.sh
@@ -101,6 +101,7 @@ export PLATFORM_SHORT_ARCH
 # Add the NDK toolchain to the PATH, needed both for contribs and for building
 # stub libraries
 NDK_TOOLCHAIN_PATH=`echo ${ANDROID_NDK}/toolchains/${PATH_HOST}-${GCCVER}/prebuilt/\`uname|tr A-Z a-z\`-*/bin`
+CHOST=${NDK_TOOLCHAIN_PATH}/${TARGET_TUPLE}
 export PATH=${NDK_TOOLCHAIN_PATH}:${PATH}
 
 ANDROID_PATH="`pwd`"
@@ -251,6 +252,7 @@ else
 fi
 
 echo "EXTRA_CFLAGS= -g ${EXTRA_CFLAGS}" >> config.mak
+echo "CHOST= ${CHOST}" >> config.mak
 export VLC_EXTRA_CFLAGS="${EXTRA_CFLAGS}"
 
 make fetch
-- 
1.8.3.4

