From 4bf6bf83ea69eeb0473fd2bae4465a2ddd179003 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 28 Jul 2012 17:21:40 +0800
Subject: [PATCH] compile.sh: support fastcompile.sh

---
 compile.sh     |   12 ++++++++++++
 fastcompile.sh |    4 ++++
 2 个文件被修改，插入 16 行(+)
 create mode 100644 fastcompile.sh

diff --git a/compile.sh b/compile.sh
index e5fa393..ad18c3a 100755
--- a/compile.sh
+++ b/compile.sh
@@ -79,6 +79,7 @@ CHOST=${NDK_TOOLCHAIN_PATH}/${TARGET_TUPLE}
 export PATH=${NDK_TOOLCHAIN_PATH}:${PATH}
 
 # 1/ libvlc, libvlccore and its plugins
+if [ -z "${VLC_FAST_COMPILE}" ]; then
 TESTED_HASH=7018018
 if [ ! -d "vlc" ]; then
     echo "VLC source not found, cloning"
@@ -107,6 +108,9 @@ EOF
         exit 1
     fi
 fi
+else
+    cd vlc
+fi
 
 if [ ${ANDROID_ABI} = "armeabi-v7a" ] ; then
     EXTRA_CFLAGS="-mfpu=vfpv3-d16 -mcpu=cortex-a8"
@@ -141,6 +145,7 @@ EXTRA_CFLAGS="${EXTRA_CFLAGS} -I${ANDROID_NDK}/sources/cxx-stl/gnu-libstdc++${CX
 echo "Building the contribs"
 mkdir -p contrib/android
 cd contrib/android
+if [ -z "${VLC_FAST_COMPILE}" ]; then
 ../bootstrap --host=${TARGET_TUPLE} --disable-disc --disable-sout --enable-small \
     --disable-dca \
     --disable-goom \
@@ -163,6 +168,7 @@ cd contrib/android
     --disable-faad2 \
     --disable-harfbuzz \
     --enable-iconv
+fi
 
 # TODO: mpeg2, theora
 
@@ -185,10 +191,13 @@ then
 MAKEFLAGS=-j`nproc`
 fi
 
+if [ -z "${VLC_FAST_COMPILE}" ]; then
 echo "EXTRA_CFLAGS= -g ${EXTRA_CFLAGS}" >> config.mak
 echo "CHOST= ${CHOST}" >> config.mak
+fi
 export VLC_EXTRA_CFLAGS="${EXTRA_CFLAGS}"
 
+if [ -z "${VLC_FAST_COMPILE}" ]; then
 if  test -d "../../../../tarballs" ; then
     echo "fetch local tarballs"
     mkdir -p ../tarballs
@@ -197,14 +206,17 @@ fi
 
 make fetch
 make $MAKEFLAGS
+fi
 
 cd ../.. && mkdir -p android && cd android
 
+if [ -z "${VLC_FAST_COMPILE}" ]; then
 echo "Bootstraping"
 ../bootstrap
 
 echo "Configuring"
 ../../configure.sh $OPTS
+fi
 
 echo "Building"
 make $MAKEFLAGS
diff --git a/fastcompile.sh b/fastcompile.sh
new file mode 100644
index 0000000..97130c1
--- /dev/null
+++ b/fastcompile.sh
@@ -0,0 +1,4 @@
+#! /bin/sh
+
+export VLC_FAST_COMPILE=1
+sh compile.sh
-- 
1.7.10.4

