From e5fa94c4376370f9bf9bfde2d1ae9e21f3c89f21 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 22 Oct 2013 15:55:01 +0800
Subject: [PATCH 01/42] .gitignore

---
 .gitignore | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/.gitignore b/.gitignore
index a9d0c62..2ca29c4 100644
--- a/.gitignore
+++ b/.gitignore
@@ -52,3 +52,8 @@ build-*
 contrib-*
 install-*
 
+# custom
+contrib/android
+pt
+intl
+android
-- 
1.9.0

