From 899ba9d90db7281930be21932f82eeebeb87475f Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 16:46:35 +0800
Subject: [PATCH] extract gecko code with 'xz -z' instead of 'tar -J'

---
 contrib/src/gecko/rules.mak |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/contrib/src/gecko/rules.mak b/contrib/src/gecko/rules.mak
index 50e4fa5..49dd12a 100644
--- a/contrib/src/gecko/rules.mak
+++ b/contrib/src/gecko/rules.mak
@@ -8,7 +8,7 @@ NPAPI_HEADERS_SVN_REV := HEAD # revision number, or just HEAD for the latest
 $(TARBALLS)/gecko-svn.tar.xz:
 	rm -Rf gecko-svn
 	$(SVN) export $(NPAPI_HEADERS_SVN_URL) gecko-svn -r $(NPAPI_HEADERS_SVN_REV)
-	tar cvJ gecko-svn > $@
+	tar cv gecko-svn | xz -z > $@
 
 .sum-gecko: gecko-svn.tar.xz
 	$(warning Integrity check skipped.)
-- 
1.7.7.4

