From d6f17086d74d2eb8b347d1e4cf01300363ee712b Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 22 Oct 2013 15:55:01 +0800
Subject: [PATCH] .gitignore

---
 .gitignore | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/.gitignore b/.gitignore
index a9d0c62..dd154ee 100644
--- a/.gitignore
+++ b/.gitignore
@@ -52,3 +52,9 @@ build-*
 contrib-*
 install-*
 
+# custom
+contrib/android
+pt
+intl
+android
+m4/
-- 
1.8.3.4

