From a9262baf05566410c941ef50ce670c4ce00bc8c9 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 19 Apr 2012 20:51:53 +0800
Subject: [PATCH 02/23] http: add http option to disable range header

---
 modules/access/http.c |   43 +++++++++++++++++++++++++++++++++++++++++--
 1 个文件被修改，插入 41 行(+)，删除 2 行(-)

diff --git a/modules/access/http.c b/modules/access/http.c
index f7a1e85765df60752365b344c4334e54ac1c07ec..2da07c2047d4173e594860618e917e547ef02da8 100644
--- a/modules/access/http.c
+++ b/modules/access/http.c
@@ -99,6 +99,10 @@ static void Close( vlc_object_t * );
 #define UA_TEXT N_("User Agent")
 #define UA_LONGTEXT N_("You can use a custom User agent or use a known one")
 
+#define HOSTS_REJECT_RANGE N_("Exclude hosts from range header detection")
+#define HOSTS_REJECT_RANGE_LONGTEXT N_(\
+    "You can disable 'range' header detection which are certern to be rejected. ")
+
 vlc_module_begin ()
     set_description( N_("HTTP input") )
     set_capability( "access", 0 )
@@ -123,6 +127,12 @@ vlc_module_begin ()
         change_safe()
     add_bool( "http-forward-cookies", true, FORWARD_COOKIES_TEXT,
               FORWARD_COOKIES_LONGTEXT, true )
+
+    add_string( "http-hosts-reject-range", NULL,
+               HOSTS_REJECT_RANGE,
+               HOSTS_REJECT_RANGE_LONGTEXT,
+           false )
+
     /* 'itpc' = iTunes Podcast */
     add_shortcut( "http", "https", "unsv", "itpc", "icyx" )
     set_callbacks( Open, Close )
@@ -145,6 +155,8 @@ struct access_sys_t
     char    *psz_referrer;
     http_auth_t auth;
 
+    char    *psz_hosts_reject_range;
+
     /* Proxy */
     bool b_proxy;
     vlc_url_t  proxy;
@@ -358,6 +370,10 @@ static int OpenWithCookies( vlc_object_t *p_this, const char *psz_access,
         }
     }
 
+    p_sys->psz_hosts_reject_range = var_InheritString( p_access, "http-hosts-reject-range" );
+    msg_Dbg( p_access, "http-hosts-reject-range" );
+    msg_Dbg( p_access, p_sys->psz_hosts_reject_range );
+
     /* HTTP referrer */
     p_sys->psz_referrer = var_InheritString( p_access, "http-referrer" );
 
@@ -1271,8 +1287,31 @@ static int Request( access_t *p_access, uint64_t i_tell )
     if( p_sys->i_version == 1 && ! p_sys->b_continuous )
     {
         p_sys->b_persist = true;
-        net_Printf( p_access, p_sys->fd, pvs,
-                    "Range: bytes=%"PRIu64"-\r\n", i_tell );
+
+        if ( p_sys->url.psz_host && *p_sys->url.psz_host &&
+             p_sys->psz_hosts_reject_range && *p_sys->psz_hosts_reject_range) {
+            bool b_reject_range = false;
+            int i_host_len = strlen( p_sys->url.psz_host );
+
+            const char* p_host = p_sys->psz_hosts_reject_range;
+            while ( p_host && *p_host ) {
+                if ( !strncasecmp(p_host, p_sys->url.psz_host, i_host_len )) {
+                    b_reject_range = true;
+                    break;
+                }
+
+                p_host = strstr( p_host, "," );
+                if ( p_host )
+                    p_host++;
+            }
+
+            if (b_reject_range) {
+                msg_Dbg( p_access, "disable range header detection" );
+            } else {
+                net_Printf( p_access, p_sys->fd, pvs,
+                "Range: bytes=%"PRIu64"-\r\n", i_tell );
+            }
+        }
         net_Printf( p_access, p_sys->fd, pvs, "Connection: close\r\n" );
     }
 
-- 
1.7.10.4

