From acb60df2fc09fb4ff495191494d30dd6bfba76a1 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 11 Aug 2012 13:07:07 +0800
Subject: [PATCH 02/32] main.mak: add CHOST for cross-compile

---
 contrib/src/main.mak |    1 +
 1 个文件被修改，插入 1 行(+)

diff --git a/contrib/src/main.mak b/contrib/src/main.mak
index fc7b877..f1012a5 100644
--- a/contrib/src/main.mak
+++ b/contrib/src/main.mak
@@ -252,6 +252,7 @@ endif
 HOSTTOOLS := \
 	CC="$(CC)" CXX="$(CXX)" LD="$(LD)" \
 	AR="$(AR)" RANLIB="$(RANLIB)" STRIP="$(STRIP)" \
+	CHOST="$(CHOST)" \
 	PATH="$(PREFIX)/bin:$(PATH)"
 HOSTVARS := $(HOSTTOOLS) \
 	CPPFLAGS="$(CPPFLAGS)" \
-- 
1.7.10.4

