From cd7fabf73cda584fe38c660a761c2be87069a3c6 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 18:21:02 +0800
Subject: [PATCH 2/5] fix toolchain config

---
 contrib/src/main.mak |    1 +
 1 file changed, 1 insertion(+)

diff --git a/contrib/src/main.mak b/contrib/src/main.mak
index 9625500..3990357 100644
--- a/contrib/src/main.mak
+++ b/contrib/src/main.mak
@@ -417,6 +417,7 @@ ifdef HAVE_DARWIN_OS
 endif
 	echo "set(CMAKE_C_COMPILER $(CC))" >> $@
 	echo "set(CMAKE_CXX_COMPILER $(CXX))" >> $@
+	echo "set(_CMAKE_TOOLCHAIN_PREFIX $(ANDROID_NDK)/toolchains/arm-linux-androideabi-4.4.3/prebuilt/darwin-x86/bin/arm-linux-androideabi-)" >> $@
 	echo "set(CMAKE_FIND_ROOT_PATH $(PREFIX))" >> $@
 	echo "set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)" >> $@
 	echo "set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)" >> $@
-- 
1.7.9.4

