From c19fadfb0b78ad223dc2aeab1703abe93c50b61c Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Fri, 8 Jun 2012 16:10:15 +0800
Subject: [PATCH 04/13] httplive: make prefetch() work with single-segment
 playlist

---
 modules/stream_filter/httplive.c |    6 +++++-
 1 个文件被修改，插入 5 行(+)，删除 1 行(-)

diff --git a/modules/stream_filter/httplive.c b/modules/stream_filter/httplive.c
index 413247a..47c84a6 100644
--- a/modules/stream_filter/httplive.c
+++ b/modules/stream_filter/httplive.c
@@ -1653,7 +1653,11 @@ static int Prefetch(stream_t *s, int *current)
         return VLC_EGENERIC;
 
     /* Download first 2 segments of this HLS stream */
-    for (int i = 0; i < 2; i++)
+    int prefetch_count = vlc_array_count(hls->segments);
+    if (prefetch_count > 2)
+        prefetch_count = 2;
+
+    for (int i = 0; i < prefetch_count; i++)
     {
         segment_t *segment = segment_GetSegment(hls, p_sys->download.segment);
         if (segment == NULL )
-- 
1.7.10.2

