From 9c840ae6fce67b09ac249a93d4fb2cba7ed4c23f Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 19 Jun 2012 17:04:09 +0800
Subject: [PATCH 06/33] core: vlc_array: support vlc_array_set

---
 include/vlc_arrays.h |    6 ++++++
 1 个文件被修改，插入 6 行(+)

diff --git a/include/vlc_arrays.h b/include/vlc_arrays.h
index c06136d..cbbc35a 100644
--- a/include/vlc_arrays.h
+++ b/include/vlc_arrays.h
@@ -373,6 +373,12 @@ vlc_array_remove( vlc_array_t * p_array, int i_index )
     }
 }
 
+static inline void
+vlc_array_set( vlc_array_t * p_array, void * p_elem, int i_index )
+{
+    p_array->pp_elems[i_index] = p_elem;
+}
+
 
 /************************************************************************
  * Dictionaries
-- 
1.7.10.4

