From 54381b84825744955b7961d660c64c60ae77255c Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 21 Jun 2012 15:44:39 +0800
Subject: [PATCH 07/32] core: src: add es_out_managed

---
 src/Makefile.am |    2 ++
 1 个文件被修改，插入 2 行(+)

diff --git a/src/Makefile.am b/src/Makefile.am
index 4e1a391..3674e4a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -46,6 +46,7 @@ pluginsinclude_HEADERS = \
 	../include/vlc_epg.h \
 	../include/vlc_es.h \
 	../include/vlc_es_out.h \
+	../include/vlc_es_out_managed.h \
 	../include/vlc_events.h \
 	../include/vlc_filter.h \
 	../include/vlc_fourcc.h \
@@ -343,6 +344,7 @@ SOURCES_libvlc_common = \
 	input/demux.c \
 	input/es_out.c \
 	input/es_out_timeshift.c \
+	input/es_out_managed.c \
 	input/event.c \
 	input/input.c \
 	input/info.h \
-- 
1.7.10.4

