From 23befa58d0479ccc7713cd6c34c0e2fd60a6dd6c Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 19 Jun 2012 17:04:09 +0800
Subject: [PATCH 07/32] core: vlc_array: support vlc_array_set

---
 include/vlc_arrays.h |    6 ++++++
 1 个文件被修改，插入 6 行(+)

diff --git a/include/vlc_arrays.h b/include/vlc_arrays.h
index 2b57fdc..7744e02 100644
--- a/include/vlc_arrays.h
+++ b/include/vlc_arrays.h
@@ -368,6 +368,12 @@ vlc_array_remove( vlc_array_t * p_array, int i_index )
     }
 }
 
+static inline void
+vlc_array_set( vlc_array_t * p_array, void * p_elem, int i_index )
+{
+    p_array->pp_elems[i_index] = p_elem;
+}
+
 
 /************************************************************************
  * Dictionaries
-- 
1.7.10.4

