From 5970471eeb357501b2a601296c04a6f4aa2b8f10 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 8 May 2012 15:20:54 +0800
Subject: [PATCH 7/7] fix video delay after clock gap for youku m3u8

---
 src/audio_output/input.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/audio_output/input.c b/src/audio_output/input.c
index 228132b..efc4e9e 100644
--- a/src/audio_output/input.c
+++ b/src/audio_output/input.c
@@ -423,7 +423,7 @@ block_t *aout_InputPlay(audio_output_t *p_aout, aout_input_t *p_input,
 
     mtime_t drift = start_date - p_buffer->i_pts;
 
-    if( drift < -i_input_rate * 3 * AOUT_MAX_PTS_ADVANCE / INPUT_RATE_DEFAULT )
+    if( drift < -i_input_rate * 3 * AOUT_MAX_PTS_ADVANCE * 10 / INPUT_RATE_DEFAULT )
     {
         msg_Warn( p_aout, "buffer way too early (%"PRId64"), clearing queue",
                   drift );
@@ -437,7 +437,7 @@ block_t *aout_InputPlay(audio_output_t *p_aout, aout_input_t *p_input,
         drift = 0;
     }
     else
-    if( drift > +i_input_rate * 3 * AOUT_MAX_PTS_DELAY / INPUT_RATE_DEFAULT )
+    if( drift > +i_input_rate * 3 * AOUT_MAX_PTS_DELAY * 10 / INPUT_RATE_DEFAULT )
     {
         msg_Warn( p_aout, "buffer way too late (%"PRId64"), dropping buffer",
                   drift );
-- 
1.7.9.4

