From 4d9b353e650d8a4e781eda4579b15c175610b70f Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 21 Jun 2012 15:47:39 +0800
Subject: [PATCH 08/27] plugin: access: add sina

---
 modules/access/Modules.am |   11 +++++++++++
 1 个文件被修改，插入 11 行(+)

diff --git a/modules/access/Modules.am b/modules/access/Modules.am
index 62703f7..5fc0a3e 100644
--- a/modules/access/Modules.am
+++ b/modules/access/Modules.am
@@ -177,6 +177,17 @@ if HAVE_XCB
 libvlc_LTLIBRARIES += libxcb_screen_plugin.la
 endif
 
+libaccess_sinahttp_plugin_la_SOURCES = sina/sinahttp.c
+libaccess_sinahttp_plugin_la_CFLAGS = $(AM_CFLAGS)
+libaccess_sinahttp_plugin_la_LIBADD = $(AM_LIBADD)
+libaccess_sinahttp_plugin_la_DEPENDENCIES =
+libvlc_LTLIBRARIES += libaccess_sinahttp_plugin.la
+
+libaccess_sinaxml_plugin_la_SOURCES = sina/sinaxml.c
+libaccess_sinaxml_plugin_la_CFLAGS = $(AM_CFLAGS) $(LIBXML2_CFLAGS)
+libaccess_sinaxml_plugin_la_LIBADD = $(AM_LIBADD) $(LIBXML2_LIBS)
+libaccess_sinaxml_plugin_la_DEPENDENCIES =
+libvlc_LTLIBRARIES += libaccess_sinaxml_plugin.la
 
 ### Digital TV ###
 
-- 
1.7.10.4

