From ed5cd13ee007b8e830a9b632a250680befe6a0eb Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 26 Jun 2012 15:47:19 +0800
Subject: [PATCH 08/30] plugin: stream_filter: add membuf

---
 modules/stream_filter/Modules.am |    6 ++++++
 1 个文件被修改，插入 6 行(+)

diff --git a/modules/stream_filter/Modules.am b/modules/stream_filter/Modules.am
index 55b22ae..8dce76b 100644
--- a/modules/stream_filter/Modules.am
+++ b/modules/stream_filter/Modules.am
@@ -13,6 +13,12 @@ libstream_filter_httplive_plugin_la_DEPENDENCIES =
 libvlc_LTLIBRARIES += libstream_filter_httplive_plugin.la
 endif
 
+libstream_filter_membuf_plugin_la_SOURCES = membuf.c
+libstream_filter_membuf_plugin_la_CFLAGS = $(AM_CFLAGS)
+libstream_filter_membuf_plugin_la_LIBADD = $(AM_LIBADD)
+libstream_filter_membuf_plugin_la_DEPENDENCIES =
+libvlc_LTLIBRARIES += libstream_filter_membuf_plugin.la
+
 if !HAVE_WIN32
 if !HAVE_WINCE
 libvlc_LTLIBRARIES += libdecomp_plugin.la
-- 
1.7.10.4

