From 7ab09cebe4e3bb6b008fba8029104e9d91a6de1c Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 19 May 2012 20:59:28 +0800
Subject: [PATCH 8/8] avcodec: disable delayed opening for packetized AAC data

---
 modules/codec/avcodec/avcodec.c |    3 ++-
 1 个文件被修改，插入 2 行(+)，删除 1 行(-)

diff --git a/modules/codec/avcodec/avcodec.c b/modules/codec/avcodec/avcodec.c
index 9a9770a..4eefc71 100644
--- a/modules/codec/avcodec/avcodec.c
+++ b/modules/codec/avcodec/avcodec.c
@@ -441,7 +441,8 @@ int ffmpeg_OpenCodec( decoder_t *p_dec )
         if( p_sys->i_codec_id == CODEC_ID_VC1 ||
             p_sys->i_codec_id == CODEC_ID_VORBIS ||
             p_sys->i_codec_id == CODEC_ID_THEORA ||
-            p_sys->i_codec_id == CODEC_ID_AAC )
+            ( p_sys->i_codec_id == CODEC_ID_AAC &&
+              !p_dec->fmt_in.b_packetized ) )
         {
             msg_Warn( p_dec, "waiting for extra data for codec %s",
                       p_sys->psz_namecodec );
-- 
1.7.10.2

