From b5813ba8e9d5d5188a72d1f82032cf23e4763db0 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 21 Jun 2012 15:47:39 +0800
Subject: [PATCH 09/29] plugin: access: add sina

---
 modules/access/Modules.am |   11 +++++++++++
 1 个文件被修改，插入 11 行(+)

diff --git a/modules/access/Modules.am b/modules/access/Modules.am
index a053f46..9e97c73 100644
--- a/modules/access/Modules.am
+++ b/modules/access/Modules.am
@@ -167,6 +167,17 @@ if HAVE_XCB
 libvlc_LTLIBRARIES += libxcb_screen_plugin.la
 endif
 
+libaccess_sinahttp_plugin_la_SOURCES = sina/sinahttp.c
+libaccess_sinahttp_plugin_la_CFLAGS = $(AM_CFLAGS)
+libaccess_sinahttp_plugin_la_LIBADD = $(AM_LIBADD)
+libaccess_sinahttp_plugin_la_DEPENDENCIES =
+libvlc_LTLIBRARIES += libaccess_sinahttp_plugin.la
+
+libaccess_sinaxml_plugin_la_SOURCES = sina/sinaxml.c
+libaccess_sinaxml_plugin_la_CFLAGS = $(AM_CFLAGS) $(LIBXML2_CFLAGS)
+libaccess_sinaxml_plugin_la_LIBADD = $(AM_LIBADD) $(LIBXML2_LIBS)
+libaccess_sinaxml_plugin_la_DEPENDENCIES =
+libvlc_LTLIBRARIES += libaccess_sinaxml_plugin.la
 
 ### Digital TV ###
 
-- 
1.7.10.4

