From a24f538e22f4eb487023e4ab5a303155dd2f14b3 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 26 Jun 2012 15:47:19 +0800
Subject: [PATCH 10/33] plugin: stream_filter: add membuf

---
 modules/stream_filter/Modules.am |    6 ++++++
 1 个文件被修改，插入 6 行(+)

diff --git a/modules/stream_filter/Modules.am b/modules/stream_filter/Modules.am
index d800cb0..c453502 100644
--- a/modules/stream_filter/Modules.am
+++ b/modules/stream_filter/Modules.am
@@ -12,6 +12,12 @@ if HAVE_GCRYPT
 libvlc_LTLIBRARIES += libhttplive_plugin.la
 endif
 
+libstream_filter_membuf_plugin_la_SOURCES = membuf.c
+libstream_filter_membuf_plugin_la_CFLAGS = $(AM_CFLAGS)
+libstream_filter_membuf_plugin_la_LIBADD = $(AM_LIBADD)
+libstream_filter_membuf_plugin_la_DEPENDENCIES =
+libvlc_LTLIBRARIES += libstream_filter_membuf_plugin.la
+
 librecord_plugin_la_SOURCES = record.c
 librecord_plugin_la_CFLAGS = $(AM_CFLAGS)
 librecord_plugin_la_LIBADD = $(AM_LIBADD)
-- 
1.7.10.4

