From bb20621c934c7952a575bac64cad1380679485e7 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 26 Jun 2012 15:47:19 +0800
Subject: [PATCH 11/34] plugin: stream_filter: add membuf

---
 modules/stream_filter/Modules.am |    6 ++++++
 1 个文件被修改，插入 6 行(+)

diff --git a/modules/stream_filter/Modules.am b/modules/stream_filter/Modules.am
index d970b3a..252ec1c 100644
--- a/modules/stream_filter/Modules.am
+++ b/modules/stream_filter/Modules.am
@@ -112,6 +112,12 @@ if HAVE_GCRYPT
 libvlc_LTLIBRARIES += libhttplive_plugin.la
 endif
 
+libstream_filter_membuf_plugin_la_SOURCES = membuf.c
+libstream_filter_membuf_plugin_la_CFLAGS = $(AM_CFLAGS)
+libstream_filter_membuf_plugin_la_LIBADD = $(AM_LIBADD)
+libstream_filter_membuf_plugin_la_DEPENDENCIES =
+libvlc_LTLIBRARIES += libstream_filter_membuf_plugin.la
+
 librecord_plugin_la_SOURCES = record.c
 librecord_plugin_la_CFLAGS = $(AM_CFLAGS)
 librecord_plugin_la_LIBADD = $(AM_LIBADD)
-- 
1.7.10.4

