From dd90acfab7022a0ca867e11bbaae28b79ef8a7d5 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 27 Jul 2013 21:04:04 +0800
Subject: [PATCH 11/36] configure.ac: add android_nativewindow

---
 configure.ac | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.ac b/configure.ac
index d1dff35..ac26523 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3216,6 +3216,8 @@ if test "${enable_android_surface}" = "yes"; then
   if test "${HAVE_ANDROID}" = "1"; then
      VLC_ADD_PLUGIN([android_surface])
      VLC_ADD_LDFLAGS([android_surface], [-ldl])
+     VLC_ADD_PLUGIN([android_nativewindow])
+     VLC_ADD_LDFLAGS([android_nativewindow], [-ldl])
   fi
 fi
 
-- 
1.8.2.1

