From 612f2182c4304bbdd5396793d04d37401550b77e Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 22 Oct 2013 14:36:09 +0800
Subject: [PATCH 11/17] plugin: stream_filter: add ringbuf, membuf

---
 modules/stream_filter/Makefile.am | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/modules/stream_filter/Makefile.am b/modules/stream_filter/Makefile.am
index 99a0647..4a426ec 100644
--- a/modules/stream_filter/Makefile.am
+++ b/modules/stream_filter/Makefile.am
@@ -118,5 +118,17 @@ if HAVE_GCRYPT
 stream_filter_LTLIBRARIES += libhttplive_plugin.la
 endif
 
+libstream_filter_ringbuf_plugin_la_SOURCES = stream_filter/ringbuf.c
+libstream_filter_ringbuf_plugin_la_CFLAGS = $(AM_CFLAGS)
+libstream_filter_ringbuf_plugin_la_LIBADD = $(AM_LIBADD)
+libstream_filter_ringbuf_plugin_la_DEPENDENCIES =
+stream_filter_LTLIBRARIES += libstream_filter_ringbuf_plugin.la
+
+libstream_filter_membuf_plugin_la_SOURCES = stream_filter/membuf.c
+libstream_filter_membuf_plugin_la_CFLAGS = $(AM_CFLAGS)
+libstream_filter_membuf_plugin_la_LIBADD = $(AM_LIBADD)
+libstream_filter_membuf_plugin_la_DEPENDENCIES =
+stream_filter_LTLIBRARIES += libstream_filter_membuf_plugin.la
+
 librecord_plugin_la_SOURCES = stream_filter/record.c
 stream_filter_LTLIBRARIES += librecord_plugin.la
-- 
2.1.2

