From e5f80a3db447fe345b3d374576c037c55bc0b481 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Wed, 11 Jul 2012 14:50:38 +0800
Subject: [PATCH 12/33] httplive: peek more bytes to detect m3u8

---
 modules/stream_filter/httplive.c |    2 +-
 1 个文件被修改，插入 1 行(+)，删除 1 行(-)

diff --git a/modules/stream_filter/httplive.c b/modules/stream_filter/httplive.c
index df185ae..ff93c79 100644
--- a/modules/stream_filter/httplive.c
+++ b/modules/stream_filter/httplive.c
@@ -169,7 +169,7 @@ static bool isHTTPLiveStreaming(stream_t *s)
 {
     const uint8_t *peek;
 
-    int size = stream_Peek(s->p_source, &peek, 46);
+    int size = stream_Peek(s->p_source, &peek, 100);
     if (size < 7)
         return false;
 
-- 
1.7.10.4

