From 563e632d3542851b2172db5b53a86e95236a5709 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 26 Jun 2012 15:47:19 +0800
Subject: [PATCH 12/33] plugin: stream_filter: add membuf

---
 modules/stream_filter/Modules.am | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/stream_filter/Modules.am b/modules/stream_filter/Modules.am
index 6df1540..f94e19b 100644
--- a/modules/stream_filter/Modules.am
+++ b/modules/stream_filter/Modules.am
@@ -118,6 +118,12 @@ libstream_filter_ringbuf_plugin_la_LIBADD = $(AM_LIBADD)
 libstream_filter_ringbuf_plugin_la_DEPENDENCIES =
 libvlc_LTLIBRARIES += libstream_filter_ringbuf_plugin.la
 
+libstream_filter_membuf_plugin_la_SOURCES = membuf.c
+libstream_filter_membuf_plugin_la_CFLAGS = $(AM_CFLAGS)
+libstream_filter_membuf_plugin_la_LIBADD = $(AM_LIBADD)
+libstream_filter_membuf_plugin_la_DEPENDENCIES =
+libvlc_LTLIBRARIES += libstream_filter_membuf_plugin.la
+
 librecord_plugin_la_SOURCES = record.c
 librecord_plugin_la_CFLAGS = $(AM_CFLAGS)
 librecord_plugin_la_LIBADD = $(AM_LIBADD)
-- 
1.8.2.1

