From 1f0a2efd91d6702dc1e8d692d3cd034dc90058ad Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 9 Oct 2012 17:53:52 +0800
Subject: [PATCH 13/32] http: trim trailing space in field contents

---
 modules/access/http.c |   15 ++++++++++++++-
 1 个文件被修改，插入 14 行(+)，删除 1 行(-)

diff --git a/modules/access/http.c b/modules/access/http.c
index cd388cc..fb5319f 100644
--- a/modules/access/http.c
+++ b/modules/access/http.c
@@ -1381,6 +1381,7 @@ static int Request( access_t *p_access, uint64_t i_tell )
     {
         char *psz = net_Gets( p_access, p_sys->fd, pvs );
         char *p;
+        char *p_trailing;
 
         if( psz == NULL )
         {
@@ -1408,7 +1409,19 @@ static int Request( access_t *p_access, uint64_t i_tell )
             goto error;
         }
         *p++ = '\0';
-        while( *p == ' ' ) p++;
+        p += strspn( p, " \t" );
+
+        /* trim trailing white space */
+        p_trailing = p + strlen( p );
+        if( p_trailing > p )
+        {
+            p_trailing--;
+            while( ( *p_trailing == ' ' || *p_trailing == '\t' ) && p_trailing > p )
+            {
+                *p_trailing = '\0';
+                p_trailing--;
+            }
+        }
 
         if( !strcasecmp( psz, "Content-Length" ) )
         {
-- 
1.7.10.4

