From 74ba7d6c2db1acd2e00d14fec05ab318d40244d5 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 27 Jul 2013 21:04:04 +0800
Subject: [PATCH] configure.ac: add android_nativewindow

---
 configure.ac | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.ac b/configure.ac
index 35bc422..d0dd3dc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3237,6 +3237,8 @@ if test "${enable_android_surface}" = "yes"; then
      VLC_ADD_LIBS([android_surface], [-ldl])
      VLC_ADD_PLUGIN([android_opaque])
      VLC_ADD_LIBS([android_opaque], [-ldl])
+     VLC_ADD_PLUGIN([android_nativewindow])
+     VLC_ADD_LIBS([android_nativewindow], [-ldl])
   fi
 fi
 
-- 
1.8.3.4

