From 8c9e2fe58f476dec717f6d44169228063fe62ad1 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 19 Apr 2012 20:51:53 +0800
Subject: [PATCH 13/32] http: add http option to disable range header

---
 modules/access/http.c |   49 +++++++++++++++++++++++++++++++++++++++++++++++--
 1 个文件被修改，插入 47 行(+)，删除 2 行(-)

diff --git a/modules/access/http.c b/modules/access/http.c
index fb5319f..5bb389a 100644
--- a/modules/access/http.c
+++ b/modules/access/http.c
@@ -102,6 +102,10 @@ static void Close( vlc_object_t * );
     "slash, e.g. FooBar/1.2.3. This option can only be specified per input " \
     "item, not globally.")
 
+#define HOSTS_REJECT_RANGE N_("Exclude hosts from range header detection")
+#define HOSTS_REJECT_RANGE_LONGTEXT N_(\
+    "You can disable 'range' header detection which are certern to be rejected. ")
+
 vlc_module_begin ()
     set_description( N_("HTTP input") )
     set_capability( "access", 0 )
@@ -126,6 +130,12 @@ vlc_module_begin ()
         change_safe()
     add_bool( "http-forward-cookies", true, FORWARD_COOKIES_TEXT,
               FORWARD_COOKIES_LONGTEXT, true )
+
+    add_string( "http-hosts-reject-range", NULL,
+               HOSTS_REJECT_RANGE,
+               HOSTS_REJECT_RANGE_LONGTEXT,
+           false )
+
     /* 'itpc' = iTunes Podcast */
     add_shortcut( "http", "https", "unsv", "itpc", "icyx" )
     set_callbacks( Open, Close )
@@ -149,6 +159,8 @@ struct access_sys_t
     char    *psz_referrer;
     http_auth_t auth;
 
+    char    *psz_hosts_reject_range;
+
     /* Proxy */
     bool b_proxy;
     vlc_url_t  proxy;
@@ -362,6 +374,13 @@ static int OpenWithCookies( vlc_object_t *p_this, const char *psz_access,
         }
     }
 
+    p_sys->psz_hosts_reject_range = var_InheritString( p_access, "http-hosts-reject-range" );
+    if( p_sys->psz_hosts_reject_range )
+    {
+        msg_Dbg( p_access, "http-hosts-reject-range" );
+        msg_Dbg( p_access, p_sys->psz_hosts_reject_range );
+    }
+
     /* HTTP referrer */
     p_sys->psz_referrer = var_InheritString( p_access, "http-referrer" );
 
@@ -1280,8 +1299,34 @@ static int Request( access_t *p_access, uint64_t i_tell )
     if( p_sys->i_version == 1 && ! p_sys->b_continuous )
     {
         p_sys->b_persist = true;
-        net_Printf( p_access, p_sys->fd, pvs,
-                    "Range: bytes=%"PRIu64"-\r\n", i_tell );
+
+        if ( p_sys->url.psz_host && *p_sys->url.psz_host ) {
+            bool b_reject_range = false;
+            if( p_sys->psz_hosts_reject_range && *p_sys->psz_hosts_reject_range )
+            {
+                int i_host_len = strlen( p_sys->url.psz_host );
+
+                const char* p_host = p_sys->psz_hosts_reject_range;
+                while ( p_host && *p_host ) {
+                    if ( !strncasecmp(p_host, p_sys->url.psz_host, i_host_len )) {
+                        b_reject_range = true;
+                        break;
+                    }
+
+                    p_host = strstr( p_host, "," );
+                    if ( p_host )
+                        p_host++;
+                }
+            }
+
+            if (b_reject_range) {
+                msg_Dbg( p_access, "disable range header detection" );
+            } else {
+                msg_Dbg( p_access, "Range: bytes=%"PRIu64"-\r\n", i_tell );
+                net_Printf( p_access, p_sys->fd, pvs,
+                "Range: bytes=%"PRIu64"-\r\n", i_tell );
+            }
+        }
         net_Printf( p_access, p_sys->fd, pvs, "Connection: close\r\n" );
     }
 
-- 
1.7.10.4

