From 9970eb23bc33ed9229d2feff8375723289c06404 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Wed, 11 Jul 2012 14:50:38 +0800
Subject: [PATCH 15/33] httplive: peek more bytes to detect m3u8

---
 modules/stream_filter/httplive.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/stream_filter/httplive.c b/modules/stream_filter/httplive.c
index 5f85e2f..465b71f 100644
--- a/modules/stream_filter/httplive.c
+++ b/modules/stream_filter/httplive.c
@@ -169,7 +169,7 @@ static bool isHTTPLiveStreaming(stream_t *s)
 {
     const uint8_t *peek;
 
-    int size = stream_Peek(s->p_source, &peek, 46);
+    int size = stream_Peek(s->p_source, &peek, 100);
     if (size < 7)
         return false;
 
-- 
1.8.2.1

