From 860579aa902b7a159da1ccf9ff9f8927327cfacf Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 24 May 2012 16:35:29 +0800
Subject: [PATCH 15/32] httplive: work with absolute path in play list

---
 modules/stream_filter/httplive.c |   23 +++++++++++++++++++++++
 1 个文件被修改，插入 23 行(+)

diff --git a/modules/stream_filter/httplive.c b/modules/stream_filter/httplive.c
index 263c2f2..a7c02eb 100644
--- a/modules/stream_filter/httplive.c
+++ b/modules/stream_filter/httplive.c
@@ -547,6 +547,29 @@ static char *relative_URI(const char *psz_url, const char *psz_path)
     if (strncmp(psz_path, "http", 4) == 0)
         return NULL;
 
+    /* Absolute path, relative to the current server */
+    if (psz_path[0] == '/')
+    {
+        char* sep = strstr(psz_url, "://");
+        if (sep)
+        {
+            sep = strchr(sep + 3, '/');
+            if (sep)
+            {
+                size_t base_length = sep - psz_url;
+                size_t path_length = strlen(psz_path);
+
+                size_t url_length = base_length + path_length;
+
+                char *psz_res = malloc(url_length + 1);
+                strncpy(psz_res, psz_url, base_length);
+                strncpy(psz_res + base_length, psz_path, path_length);
+                psz_res[url_length] = 0;
+                return psz_res;
+            }
+        }
+    }
+
     char    *path_end = strrchr(psz_url, '/');
     if (path_end == NULL)
         return NULL;
-- 
1.7.10.4

