From 9d0d0b425734f942b36feee050345e3f26ac1ccd Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Wed, 11 Jul 2012 14:50:38 +0800
Subject: [PATCH 16/35] httplive: peek more bytes to detect m3u8

---
 modules/stream_filter/httplive.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/stream_filter/httplive.c b/modules/stream_filter/httplive.c
index 725a538..a6e4d3e 100644
--- a/modules/stream_filter/httplive.c
+++ b/modules/stream_filter/httplive.c
@@ -181,7 +181,7 @@ static bool isHTTPLiveStreaming(stream_t *s)
 {
     const uint8_t *peek;
 
-    int size = stream_Peek(s->p_source, &peek, 46);
+    int size = stream_Peek(s->p_source, &peek, 100);
     if (size < 7)
         return false;
 
-- 
1.8.3.4

