From c556a3c4045033a24430abb3203fb8025fdc7402 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 18 Jun 2012 16:41:22 +0800
Subject: [PATCH 17/20] sina: support sina v_play

---
 modules/access/sina.c |    2 +-
 1 个文件被修改，插入 1 行(+)，删除 1 行(-)

diff --git a/modules/access/sina.c b/modules/access/sina.c
index 83953ac..727d5dd 100644
--- a/modules/access/sina.c
+++ b/modules/access/sina.c
@@ -1,5 +1,5 @@
 /*****************************************************************************
- * sina.c: Sina Video support (uncrypted)
+ * sina.c: Sina Video support
  *****************************************************************************
  *
  * Authors: Rui Zhang <bbcallen _AT_ gmail _DOT_ com>
-- 
1.7.10.2

