From 9fc662d83aca36ea007d86cd58429b2bcfddb9d8 Mon Sep 17 00:00:00 2001
From: Zhang Rui <bbcallen@gmail.com>
Date: Thu, 13 Feb 2014 12:41:38 +0800
Subject: [PATCH 19/21] mediacodec: echo name

---
 modules/codec/omxil/android_mediacodec.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/modules/codec/omxil/android_mediacodec.c b/modules/codec/omxil/android_mediacodec.c
index 44184c6..1313b12 100644
--- a/modules/codec/omxil/android_mediacodec.c
+++ b/modules/codec/omxil/android_mediacodec.c
@@ -557,6 +557,15 @@ static int OpenDecoder(vlc_object_t *p_this)
     if (!p_sys->timestamp_fifo)
         goto error;
 
+    if (p_sys->direct_rendering) {
+        char *psz_tmp = NULL;
+        asprintf( &psz_tmp, "%s, SurfaceRender", p_sys->name );
+        var_SetString( p_dec, "module-video-decoder-impl", psz_tmp );
+        free( psz_tmp );
+    } else {
+        var_SetString( p_dec, "module-video-decoder-impl", p_sys->name );
+    }
+
     return VLC_SUCCESS;
 
  error:
-- 
2.1.3

