From 70394c136d9385a684178ca0d8b64678b61ee48e Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 29 Jul 2012 16:44:03 +0800
Subject: [PATCH 20/37] httplive: retry segment if failed

---
 modules/stream_filter/httplive.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/modules/stream_filter/httplive.c b/modules/stream_filter/httplive.c
index 41fd84b..fdac815 100644
--- a/modules/stream_filter/httplive.c
+++ b/modules/stream_filter/httplive.c
@@ -1845,7 +1845,14 @@ static int hls_Download(stream_t *s, segment_t *segment)
         vlc_cond_wait(&p_sys->wait, &p_sys->lock);
     vlc_mutex_unlock(&p_sys->lock);
 
-    stream_t *p_ts = stream_UrlNew(s, segment->url);
+    stream_t *p_ts = NULL;
+    for( int i = 0; i < 3; ++i )
+    {
+        /* retry or may stalling */
+        p_ts = stream_UrlNew(s, segment->url);
+        if( p_ts )
+            break;
+    }
     if (p_ts == NULL)
         return VLC_EGENERIC;
 
-- 
1.8.3.4

