From 24ab26475e2a07f08af0c7999526ddd8a7c82194 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 21 Jun 2012 15:51:06 +0800
Subject: [PATCH 22/23] plugin_stream_filter: add memory_buffer

---
 modules/stream_filter/Modules.am |    6 ++++++
 1 个文件被修改，插入 6 行(+)

diff --git a/modules/stream_filter/Modules.am b/modules/stream_filter/Modules.am
index 78efb7d946f21de817902f81405aeb7d9782895c..f11e77e1176f559c444db8b10349f008ef0b335a 100644
--- a/modules/stream_filter/Modules.am
+++ b/modules/stream_filter/Modules.am
@@ -15,6 +15,12 @@ libstream_filter_httplive_plugin_la_DEPENDENCIES =
 libvlc_LTLIBRARIES += libstream_filter_httplive_plugin.la
 endif
 
+libstream_filter_memory_buffer_plugin_la_SOURCES = memory_buffer.c
+libstream_filter_memory_buffer_plugin_la_CFLAGS = $(AM_CFLAGS)
+libstream_filter_memory_buffer_plugin_la_LIBADD = $(AM_LIBADD)
+libstream_filter_memory_buffer_plugin_la_DEPENDENCIES =
+libvlc_LTLIBRARIES += libstream_filter_memory_buffer_plugin.la
+
 if !HAVE_WIN32
 if !HAVE_WINCE
 libvlc_LTLIBRARIES += libdecomp_plugin.la
-- 
1.7.10.4

