From 8bcdf28c0bce513cd8efa0e768f2f5d1327a59b0 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 21 Jun 2012 00:36:52 +0800
Subject: [PATCH 20/20] sina: use memory_buffer

---
 modules/access/sina.c |   11 +++++++++++
 1 个文件被修改，插入 11 行(+)

diff --git a/modules/access/sina.c b/modules/access/sina.c
index 727d5dd..850ca0e 100644
--- a/modules/access/sina.c
+++ b/modules/access/sina.c
@@ -617,6 +617,17 @@ static int sina_AssureOpenSegment( demux_t *p_demux, int i_segment )
                 continue;
             }
 
+            msg_Dbg( p_demux, "sina open filter for segment stream %d", p_new_segment->i_order );
+            stream_t *p_stream_filter = stream_FilterNew( p_stream, "memory_buffer" );
+            if( !p_stream_filter )
+            {
+                msg_Err( p_demux, "failed to open stream_filter %s", p_new_segment->p_url );
+            }
+            else
+            {
+                p_stream = p_stream_filter;
+            }
+
             int64_t segment_size = stream_Size( p_stream );
             msg_Info( p_demux, "sina segment size %"PRId64, segment_size );
 
-- 
1.7.10.2

