From 9eb7d737bdce332f80c2696019c7d377c8f6e1af Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 28 Jul 2012 21:35:51 +0800
Subject: [PATCH 23/33] httplive: fix crash

---
 modules/stream_filter/httplive.c |    1 +
 1 个文件被修改，插入 1 行(+)

diff --git a/modules/stream_filter/httplive.c b/modules/stream_filter/httplive.c
index 75c5c61..5bcd623 100644
--- a/modules/stream_filter/httplive.c
+++ b/modules/stream_filter/httplive.c
@@ -1360,6 +1360,7 @@ static int hls_UpdatePlaylist(stream_t *s, hls_stream_t *hls_new, hls_stream_t *
     msg_Info(s, "updating hls stream (program-id=%d, bandwidth=%"PRIu64") has %d segments",
              hls_new->id, hls_new->bandwidth, count);
 
+    vlc_mutex_lock(&hls_old->lock);
     bool b_append_new_segment = false;
     for (int n = 0; n < count; n++)
     {
-- 
1.7.10.4

