From aa0cee5082c2220d693608f663503a1fd2ae75d3 Mon Sep 17 00:00:00 2001
From: Zhang Rui <bbcallen@gmail.com>
Date: Sun, 4 Jan 2015 15:41:02 +0800
Subject: [PATCH 25/25] MediaCodec: do not drop packet

---
 modules/codec/omxil/android_mediacodec.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/modules/codec/omxil/android_mediacodec.c b/modules/codec/omxil/android_mediacodec.c
index 161b7a4..926c8f8 100644
--- a/modules/codec/omxil/android_mediacodec.c
+++ b/modules/codec/omxil/android_mediacodec.c
@@ -954,7 +954,12 @@ static picture_t *DecodeVideo(decoder_t *p_dec, block_t **pp_block)
 
     jni_attach_thread(&env, THREAD_NAME);
 
-    if (p_block->i_flags & (BLOCK_FLAG_DISCONTINUITY|BLOCK_FLAG_CORRUPTED)) {
+    /* XXX: this is really a flush block from
+     +       DecoderBlockFlushNew() */
+    if ((p_block->i_flags & (BLOCK_FLAG_DISCONTINUITY|BLOCK_FLAG_CORRUPTED)) &&
+        (p_block->i_buffer == 128) &&
+        (p_block->i_dts == VLC_TS_INVALID) &&
+        (p_block->i_pts == VLC_TS_INVALID)) {
         block_Release(p_block);
         timestamp_FifoEmpty(p_sys->timestamp_fifo);
         if (p_sys->decoded) {
-- 
2.0.0

