From f1cf4254f62095cc0e0b0a7fcca87d4232f30678 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 22 Jul 2012 18:43:17 +0800
Subject: [PATCH 26/42] omxil: disable debug info

---
 modules/codec/omxil/omxil.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/codec/omxil/omxil.c b/modules/codec/omxil/omxil.c
index 5a28fbd..d636d46 100644
--- a/modules/codec/omxil/omxil.c
+++ b/modules/codec/omxil/omxil.c
@@ -42,7 +42,7 @@
 #include "OMX_Broadcom.h"
 
 #ifndef NDEBUG
-# define OMXIL_EXTRA_DEBUG
+//# define OMXIL_EXTRA_DEBUG
 #endif
 
 #define SENTINEL_FLAG 0x10000
-- 
1.9.0

