From 48195c8ed1828529cf6cc85ed5211ee5a1cee21b Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Wed, 4 Jul 2012 18:28:44 +0800
Subject: [PATCH 27/34] omxil: disable audio codec

---
 modules/codec/omxil/omxil.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/codec/omxil/omxil.c b/modules/codec/omxil/omxil.c
index e2a35fe..013cc05 100644
--- a/modules/codec/omxil/omxil.c
+++ b/modules/codec/omxil/omxil.c
@@ -1009,6 +1009,9 @@ static int OpenDecoder( vlc_object_t *p_this )
     if( 0 || !GetOmxRole(p_dec->fmt_in.i_codec, p_dec->fmt_in.i_cat, false) )
         return VLC_EGENERIC;
 
+    if( p_dec->fmt_in.i_cat != VIDEO_ES )
+        return VLC_EGENERIC;
+
     status = OpenGeneric( p_this, false );
     if(status != VLC_SUCCESS) return status;
 
-- 
1.8.3.4

