From d9bba81888b661e13cc4b3dd3eff721b50eb2d51 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Wed, 4 Jul 2012 18:27:53 +0800
Subject: [PATCH 27/32] omxil: use pts for OMX.SEC.AVC.Decoder

---
 modules/codec/omxil/omxil.c |    2 ++
 1 个文件被修改，插入 2 行(+)

diff --git a/modules/codec/omxil/omxil.c b/modules/codec/omxil/omxil.c
index e2d246f..55d3b48 100644
--- a/modules/codec/omxil/omxil.c
+++ b/modules/codec/omxil/omxil.c
@@ -1295,6 +1295,8 @@ loaded:
     p_dec->b_need_packetized = true;
     if (!strcmp(p_sys->psz_component, "OMX.TI.DUCATI1.VIDEO.DECODER"))
         p_sys->b_use_pts = 1;
+    if (!strcmp(p_sys->psz_component, "OMX.SEC.AVC.Decoder"))
+        p_sys->b_use_pts = 1;
 
     if( p_dec->fmt_in.i_cat != VIDEO_ES )
         var_SetString( p_dec, "module-audio-decoder-impl", p_sys->psz_component );
-- 
1.7.10.4

