From 261ea1d1db36d5e9046fe56db85feb94a1d1d149 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 22 Jul 2012 18:43:17 +0800
Subject: [PATCH 28/33] omxil: disable debug info

---
 modules/codec/omxil/omxil.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/codec/omxil/omxil.c b/modules/codec/omxil/omxil.c
index 1d8f5ac..66eed03 100644
--- a/modules/codec/omxil/omxil.c
+++ b/modules/codec/omxil/omxil.c
@@ -41,7 +41,7 @@
 #include "omxil_core.h"
 
 #ifndef NDEBUG
-# define OMXIL_EXTRA_DEBUG
+//# define OMXIL_EXTRA_DEBUG
 #endif
 
 #define SENTINEL_FLAG 0x10000
-- 
1.8.2.1

