From 508f84872c4876e153c85519b9a2c07c2daf759d Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Wed, 4 Jul 2012 18:28:44 +0800
Subject: [PATCH 28/32] omxil: disable audio codec

---
 modules/codec/omxil/omxil.c |    2 ++
 1 个文件被修改，插入 2 行(+)

diff --git a/modules/codec/omxil/omxil.c b/modules/codec/omxil/omxil.c
index 55d3b48..70008c4 100644
--- a/modules/codec/omxil/omxil.c
+++ b/modules/codec/omxil/omxil.c
@@ -1024,6 +1024,8 @@ static int OpenDecoder( vlc_object_t *p_this )
     if( p_dec->fmt_in.i_cat != VIDEO_ES && !p_dec->b_force)
         return VLC_EGENERIC;
 #endif
+    if( p_dec->fmt_in.i_cat != VIDEO_ES )
+        return VLC_EGENERIC;
 
     status = OpenGeneric( p_this, false );
     if(status != VLC_SUCCESS) return status;
-- 
1.7.10.4

