From a998acf8218847a61a53d260b0ec2a82c1d4c15d Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Wed, 23 Oct 2013 21:22:05 +0800
Subject: [PATCH 28/35] mediacodec: disable OMX.MTK.VIDEO.DECODER.AVC

---
 modules/codec/omxil/android_mediacodec.c | 22 +++++++++++++++-------
 1 file changed, 15 insertions(+), 7 deletions(-)

diff --git a/modules/codec/omxil/android_mediacodec.c b/modules/codec/omxil/android_mediacodec.c
index 499f1f5..fe0832f 100644
--- a/modules/codec/omxil/android_mediacodec.c
+++ b/modules/codec/omxil/android_mediacodec.c
@@ -306,13 +306,21 @@ static int OpenDecoder(vlc_object_t *p_this)
             jobject name = (*env)->CallObjectMethod(env, info, p_sys->get_name);
             jsize name_len = (*env)->GetStringUTFLength(env, name);
             const char *name_ptr = (*env)->GetStringUTFChars(env, name, NULL);
-            msg_Dbg(p_dec, "using %.*s", name_len, name_ptr);
-            p_sys->name = malloc(name_len + 1);
-            memcpy(p_sys->name, name_ptr, name_len);
-            p_sys->name[name_len] = '\0';
-            (*env)->ReleaseStringUTFChars(env, name, name_ptr);
-            codec_name = name;
-            break;
+            if (name_len == 0 || name_ptr == NULL) {
+                msg_Dbg(p_dec, "invalid codec name %.*s", name_len, name_ptr);
+            } else if (0 == strncmp(name_ptr, "OMX.MTK.VIDEO.DECODER.AVC", name_len)) {
+                msg_Dbg(p_dec, "disable %.*s", name_len, name_ptr);
+                (*env)->ReleaseStringUTFChars(env, name, name_ptr);
+                (*env)->DeleteLocalRef(env, name);
+            } else {
+                msg_Dbg(p_dec, "using %.*s", name_len, name_ptr);
+                p_sys->name = malloc(name_len + 1);
+                memcpy(p_sys->name, name_ptr, name_len);
+                p_sys->name[name_len] = '\0';
+                (*env)->ReleaseStringUTFChars(env, name, name_ptr);
+                codec_name = name;
+                break;
+            }
         }
         (*env)->DeleteLocalRef(env, info);
     }
-- 
1.8.3.4

