From 1d65c7212224f3f19ab1f5615ad19d3f57fc73bc Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 12 Jul 2012 19:59:25 +0800
Subject: [PATCH 29/29] avformat: avoid EOF if av_read_frame returns
 AVERROR(EAGAIN)

---
 modules/demux/avformat/demux.c |    7 ++++++-
 1 个文件被修改，插入 6 行(+)，删除 1 行(-)

diff --git a/modules/demux/avformat/demux.c b/modules/demux/avformat/demux.c
index cce867e..21a0bf7 100644
--- a/modules/demux/avformat/demux.c
+++ b/modules/demux/avformat/demux.c
@@ -591,8 +591,13 @@ static int Demux( demux_t *p_demux )
     int64_t     i_start_time;
 
     /* Read a frame */
-    if( av_read_frame( p_sys->ic, &pkt ) )
+    int i_av_ret = av_read_frame( p_sys->ic, &pkt );
+    if( i_av_ret )
     {
+        /* Avoid EOF if av_read_frame returns AVERROR(EAGAIN) */
+        if( i_av_ret == AVERROR(EAGAIN) )
+            return 1;
+
         return 0;
     }
     if( pkt.stream_index < 0 || pkt.stream_index >= p_sys->i_tk )
-- 
1.7.10.4

