From e44ad5cf22a898164d8159cdef162289ea7c8551 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Wed, 4 Jul 2012 18:28:44 +0800
Subject: [PATCH 29/36] omxil: disable audio codec

---
 modules/codec/omxil/omxil.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/codec/omxil/omxil.c b/modules/codec/omxil/omxil.c
index a0831f9..1d8f5ac 100644
--- a/modules/codec/omxil/omxil.c
+++ b/modules/codec/omxil/omxil.c
@@ -939,6 +939,9 @@ static int OpenDecoder( vlc_object_t *p_this )
     if( 0 || !GetOmxRole(p_dec->fmt_in.i_codec, p_dec->fmt_in.i_cat, false) )
         return VLC_EGENERIC;
 
+    if( p_dec->fmt_in.i_cat != VIDEO_ES )
+        return VLC_EGENERIC;
+
     status = OpenGeneric( p_this, false );
     if(status != VLC_SUCCESS) return status;
 
-- 
1.8.2.1

