From 685d249617674185f070757b33f60ea931b6fd40 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 22 Jul 2012 18:43:17 +0800
Subject: [PATCH 30/33] omxil: disable debug info

---
 modules/codec/omxil/omxil.c |    2 +-
 1 个文件被修改，插入 1 行(+)，删除 1 行(-)

diff --git a/modules/codec/omxil/omxil.c b/modules/codec/omxil/omxil.c
index 4f2928a..691c1c6 100644
--- a/modules/codec/omxil/omxil.c
+++ b/modules/codec/omxil/omxil.c
@@ -51,7 +51,7 @@
 #include "omxil.h"
 
 #ifndef NDEBUG
-# define OMXIL_EXTRA_DEBUG
+//# define OMXIL_EXTRA_DEBUG
 #endif
 
 #define SENTINEL_FLAG 0x10000
-- 
1.7.10.4

