From 23fe5de148561aafcdd7caf295a58cc7d9bb45b3 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 24 Oct 2013 16:29:21 +0800
Subject: [PATCH 32/35] avcodec: do not align dimensions on android

---
 modules/codec/avcodec/video.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/codec/avcodec/video.c b/modules/codec/avcodec/video.c
index fea22bb..0341e3a 100644
--- a/modules/codec/avcodec/video.c
+++ b/modules/codec/avcodec/video.c
@@ -129,12 +129,14 @@ static inline picture_t *ffmpeg_NewPictBuf( decoder_t *p_dec,
     int width = p_context->coded_width;
     int height = p_context->coded_height;
 
+#ifndef __ANDROID__
     if( p_sys->p_va == NULL )
     {
         int aligns[AV_NUM_DATA_POINTERS];
 
         avcodec_align_dimensions2(p_context, &width, &height, aligns);
     }
+#endif
 
     if( width == 0 || height == 0)
         return NULL; /* invalid display size */
-- 
1.8.3.4

