From 1bde1ecd2a65ea39896ff1d49e52341399a3fed6 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 26 Feb 2013 17:25:39 +0800
Subject: [PATCH 32/33] androidsurface: setup format depending on surface
 format

---
 modules/video_output/androidsurface.c |   56 +++++++++++++++++++++++++++++++--
 1 个文件被修改，插入 54 行(+)，删除 2 行(-)

diff --git a/modules/video_output/androidsurface.c b/modules/video_output/androidsurface.c
index 6784ab9..e0b1fb9 100644
--- a/modules/video_output/androidsurface.c
+++ b/modules/video_output/androidsurface.c
@@ -123,6 +123,7 @@ struct picture_sys_t {
 
 static int  AndroidLockSurface(picture_t *);
 static void AndroidUnlockSurface(picture_t *);
+static int  AndroidRetrieveSurfaceFormat( vout_display_sys_t* );
 
 static vlc_mutex_t single_instance = VLC_STATIC_MUTEX;
 
@@ -160,6 +161,13 @@ static void *InitLibrary(vout_display_sys_t *sys)
     return NULL;
 }
 
+// android.graphics.PixelFormat.RGB_565
+#define ANDROID_PIXEL_FORMAT_RGB_565    0x4
+// android.graphics.PixelFormat.RGBX_8888
+#define ANDROID_PIXEL_FORMAT_RGBX_8888  0x2
+// android.graphics.ImageFormat.YV12
+#define ANDROID_IMAGE_FORMAT_YV12 0x32315659
+
 static int Open(vlc_object_t *p_this)
 {
     vout_display_t *vd = (vout_display_t *)p_this;
@@ -193,8 +201,27 @@ static int Open(vlc_object_t *p_this)
     if( psz_fcc ) {
         fmt.i_chroma = vlc_fourcc_GetCodecFromString(VIDEO_ES, psz_fcc);
         free(psz_fcc);
-    } else
-        fmt.i_chroma = VLC_CODEC_RGB32;
+    } else {
+        int surface_format = AndroidRetrieveSurfaceFormat( sys );
+        switch( surface_format ) {
+            case ANDROID_IMAGE_FORMAT_YV12:
+                fmt.i_chroma = VLC_CODEC_YV12;
+                break;
+
+            case ANDROID_PIXEL_FORMAT_RGB_565:
+                fmt.i_chroma = VLC_CODEC_RGB16;
+                break;
+
+            case ANDROID_PIXEL_FORMAT_RGBX_8888:
+                fmt.i_chroma = VLC_CODEC_RGB32;
+                break;
+
+            default:
+                // FIXME: supposed to be RGBX_8888
+                fmt.i_chroma = VLC_CODEC_RGB32;
+                break;
+        }
+    }
 
     switch(fmt.i_chroma) {
         case VLC_CODEC_YV12:
@@ -296,6 +323,31 @@ static picture_pool_t *Pool(vout_display_t *vd, unsigned count)
     return vd->sys->pool;
 }
 
+static int AndroidRetrieveSurfaceFormat( vout_display_sys_t* p_sys )
+{
+    SurfaceInfo surface_info;
+    void* p_surface;
+
+    p_surface = jni_LockAndGetAndroidSurface( );
+    if( unlikely( !p_surface ) )
+    {
+        jni_UnlockAndroidSurface( );
+        return VLC_EGENERIC;
+    }
+
+    if (p_sys->s_lock)
+        p_sys->s_lock(p_surface, &surface_info, 1);
+    else
+        p_sys->s_lock2(p_surface, &surface_info, NULL);
+
+    int surface_format = surface_info.format;
+
+    p_sys->s_unlockAndPost( p_surface );
+    jni_UnlockAndroidSurface( );
+
+    return surface_format;
+}
+
 #define ALIGN_16_PIXELS( x ) ( ( ( x ) + 15 ) / 16 * 16 )
 static void SetupPictureYV12( SurfaceInfo* p_surfaceInfo, picture_t *p_picture )
 {
-- 
1.7.10.4

