From 72f501b9e81acd054c778af29df64097a05c779b Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 26 Feb 2013 17:25:39 +0800
Subject: [PATCH 34/35] androidsurface: setup format depending on surface
 format

---
 modules/video_output/androidsurface.c | 60 +++++++++++++++++++++++++++++++++--
 1 file changed, 58 insertions(+), 2 deletions(-)

diff --git a/modules/video_output/androidsurface.c b/modules/video_output/androidsurface.c
index d5a9c39..a5b4723 100644
--- a/modules/video_output/androidsurface.c
+++ b/modules/video_output/androidsurface.c
@@ -124,6 +124,9 @@ struct vout_display_sys_t {
     jobject jsurf;
     ANativeWindow *window;
 
+    picture_resource_t resource;
+    vout_display_t *vd;
+
     /* density */
     int i_sar_num;
     int i_sar_den;
@@ -140,6 +143,7 @@ struct picture_sys_t {
 
 static int  AndroidLockSurface(picture_t *);
 static void AndroidUnlockSurface(picture_t *);
+static int  AndroidRetrieveSurfaceFormat( vout_display_sys_t* );
 
 static vlc_mutex_t single_instance = VLC_STATIC_MUTEX;
 
@@ -204,6 +208,13 @@ static void *InitLibrary2(vout_display_sys_t *sys)
     return NULL;
 }
 
+// android.graphics.PixelFormat.RGB_565
+#define ANDROID_PIXEL_FORMAT_RGB_565    0x4
+// android.graphics.PixelFormat.RGBX_8888
+#define ANDROID_PIXEL_FORMAT_RGBX_8888  0x2
+// android.graphics.ImageFormat.YV12
+#define ANDROID_IMAGE_FORMAT_YV12 0x32315659
+
 static int Open(vlc_object_t *p_this)
 {
     vout_display_t *vd = (vout_display_t *)p_this;
@@ -222,6 +233,7 @@ static int Open(vlc_object_t *p_this)
     }
 
     /* */
+    sys->vd = vd;
     sys->p_library = InitLibrary2(sys);
     if (!sys->p_library)
         sys->p_library = InitLibrary(sys);
@@ -239,8 +251,27 @@ static int Open(vlc_object_t *p_this)
     if( psz_fcc ) {
         fmt.i_chroma = vlc_fourcc_GetCodecFromString(VIDEO_ES, psz_fcc);
         free(psz_fcc);
-    } else
-        fmt.i_chroma = VLC_CODEC_RGB32;
+    } else {
+        int surface_format = AndroidRetrieveSurfaceFormat( sys );
+        switch( surface_format ) {
+            case ANDROID_IMAGE_FORMAT_YV12:
+                fmt.i_chroma = VLC_CODEC_YV12;
+                break;
+
+            case ANDROID_PIXEL_FORMAT_RGB_565:
+                fmt.i_chroma = VLC_CODEC_RGB16;
+                break;
+
+            case ANDROID_PIXEL_FORMAT_RGBX_8888:
+                fmt.i_chroma = VLC_CODEC_RGB32;
+                break;
+
+            default:
+                // FIXME: supposed to be RGBX_8888
+                fmt.i_chroma = VLC_CODEC_RGB32;
+                break;
+        }
+    }
 
     switch(fmt.i_chroma) {
         case VLC_CODEC_YV12:
@@ -341,6 +372,31 @@ static picture_pool_t *Pool(vout_display_t *vd, unsigned count)
     return vd->sys->pool;
 }
 
+static int AndroidRetrieveSurfaceFormat( vout_display_sys_t* p_sys )
+{
+    SurfaceInfo surface_info;
+    void* p_surface;
+
+    p_surface = jni_LockAndGetAndroidSurface( );
+    if( unlikely( !p_surface ) )
+    {
+        jni_UnlockAndroidSurface( );
+        return VLC_EGENERIC;
+    }
+
+    if (p_sys->s_lock)
+        p_sys->s_lock(p_surface, &surface_info, 1);
+    else
+        p_sys->s_lock2(p_surface, &surface_info, NULL);
+
+    int surface_format = surface_info.format;
+
+    p_sys->s_unlockAndPost( p_surface );
+    jni_UnlockAndroidSurface( );
+
+    return surface_format;
+}
+
 #define ALIGN_16_PIXELS( x ) ( ( ( x ) + 15 ) / 16 * 16 )
 static void SetupPictureYV12( SurfaceInfo* p_surfaceInfo, picture_t *p_picture )
 {
-- 
1.8.3.4

