From bfbb31b1808b9a75dfadbd3d8e829238c9da9660 Mon Sep 17 00:00:00 2001
From: Zhang Rui <bbcallen@gmail.com>
Date: Thu, 13 Feb 2014 12:41:38 +0800
Subject: [PATCH 36/37] MediaCodec: echo name

---
 modules/codec/omxil/android_mediacodec.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/modules/codec/omxil/android_mediacodec.c b/modules/codec/omxil/android_mediacodec.c
index cbca603..fe48bc3 100644
--- a/modules/codec/omxil/android_mediacodec.c
+++ b/modules/codec/omxil/android_mediacodec.c
@@ -459,6 +459,15 @@ static int OpenDecoder(vlc_object_t *p_this)
         goto error;
     (*env)->DeleteLocalRef(env, format);
 
+    if (p_sys->direct_rendering) {
+        char *psz_tmp = NULL;
+        asprintf( &psz_tmp, "%s, SurfaceRender", p_sys->name );
+        var_SetString( p_dec, "module-video-decoder-impl", psz_tmp );
+        free( psz_tmp );
+    } else {
+        var_SetString( p_dec, "module-video-decoder-impl", p_sys->name );
+    }
+
     return VLC_SUCCESS;
 
  error:
-- 
1.8.3.4

