From 1ab72527c896403a2d84f7823e5d3776367bc759 Mon Sep 17 00:00:00 2001
From: Zhang Rui <bbcallen@gmail.com>
Date: Wed, 19 Feb 2014 16:34:32 +0800
Subject: [PATCH 38/42] MediaCodec: check exception before release buffer

---
 modules/codec/omxil/android_mediacodec.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules/codec/omxil/android_mediacodec.c b/modules/codec/omxil/android_mediacodec.c
index 564d473..6cbfcf9 100644
--- a/modules/codec/omxil/android_mediacodec.c
+++ b/modules/codec/omxil/android_mediacodec.c
@@ -760,6 +760,11 @@ static void DisplayBuffer(picture_sys_t* p_picsys, bool b_render)
     /* Release the MediaCodec buffer. */
     JNIEnv *env = NULL;
     vlcjni_setup_thread_env(&env);
+    if ((*env)->ExceptionOccurred(env)) {
+        msg_Err(p_dec, "Exception before MediaCodec.releaseOutputBuffer (DisplayBuffer)");
+        (*env)->ExceptionClear(env);
+    }
+
     (*env)->CallVoidMethod(env, p_sys->codec, p_sys->release_output_buffer, i_index, b_render);
     if ((*env)->ExceptionOccurred(env)) {
         msg_Err(p_dec, "Exception in MediaCodec.releaseOutputBuffer (DisplayBuffer)");
-- 
1.9.0

