From 7ca5cc22ea9c611089aa3d788e72d1d776fcaf83 Mon Sep 17 00:00:00 2001
From: Zhang Rui <bbcallen@gmail.com>
Date: Wed, 19 Feb 2014 16:34:32 +0800
Subject: [PATCH 39/39] MediaCodec: check exception before release buffer

---
 modules/codec/omxil/android_mediacodec.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules/codec/omxil/android_mediacodec.c b/modules/codec/omxil/android_mediacodec.c
index 5b10b74..c3bad9e 100644
--- a/modules/codec/omxil/android_mediacodec.c
+++ b/modules/codec/omxil/android_mediacodec.c
@@ -628,6 +628,11 @@ static void DisplayBuffer(picture_sys_t* p_picsys, bool b_render)
     /* Release the MediaCodec buffer. */
     JNIEnv *env = NULL;
     vlcjni_setup_thread_env(&env);
+    if ((*env)->ExceptionOccurred(env)) {
+        msg_Err(p_dec, "Exception before MediaCodec.releaseOutputBuffer (DisplayBuffer)");
+        (*env)->ExceptionClear(env);
+    }
+
     (*env)->CallVoidMethod(env, p_sys->codec, p_sys->release_output_buffer, i_index, b_render);
     if ((*env)->ExceptionOccurred(env)) {
         msg_Err(p_dec, "Exception in MediaCodec.releaseOutputBuffer (DisplayBuffer)");
-- 
1.8.3.4

