From e53571e0ada680949cc45df75ac176098a2a3f2b Mon Sep 17 00:00:00 2001
From: Zhang Rui <bbcallen@gmail.com>
Date: Tue, 25 Feb 2014 11:48:25 +0800
Subject: [PATCH 40/42] MediaCodec: do not drop packet

---
 modules/codec/omxil/android_mediacodec.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/modules/codec/omxil/android_mediacodec.c b/modules/codec/omxil/android_mediacodec.c
index 7c69643..7262ca4 100644
--- a/modules/codec/omxil/android_mediacodec.c
+++ b/modules/codec/omxil/android_mediacodec.c
@@ -902,7 +902,12 @@ static picture_t *DecodeVideo(decoder_t *p_dec, block_t **pp_block)
 
     vlcjni_setup_thread_env(&env);
 
-    if (p_block->i_flags & (BLOCK_FLAG_DISCONTINUITY|BLOCK_FLAG_CORRUPTED)) {
+    /* XXX: this is really a flush block from DecoderBlockFlushNew() */
+    msg_Err( p_dec, "%x %x %x %x", p_block->i_flags, p_block->i_buffer, p_block->i_dts, p_block->i_pts);
+    if ((p_block->i_flags & (BLOCK_FLAG_DISCONTINUITY|BLOCK_FLAG_CORRUPTED)) &&
+        (p_block->i_buffer == 128) &&
+        (p_block->i_dts == VLC_TS_INVALID) &&
+        (p_block->i_pts == VLC_TS_INVALID)) {
         block_Release(p_block);
         if (p_sys->decoded) {
             /* Invalidate all pictures that are currently in flight
-- 
1.8.5.3

