From 225fa3c93b278808db78dc32ede521fb1d20b925 Mon Sep 17 00:00:00 2001
From: Ming Hu <tewilove@gmail.com>
Date: Wed, 9 Apr 2014 11:36:36 +0800
Subject: [PATCH 42/42] configure: vorbis is not mandatory

---
 configure.ac | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3d1b67a..4ef6f89 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2000,9 +2000,11 @@ dnl
 dnl  ogg demux plugin
 dnl
 PKG_ENABLE_MODULES_VLC([OGG], [], [ogg >= 1.0], [Ogg demux support], [auto])
-PKG_CHECK_MODULES(LIBVORBIS, [vorbis >= 1.1], [
-    AC_DEFINE(HAVE_LIBVORBIS, 1, [Define to 1 if you have the libvorbis])
-],[])
+if test "x${enable_ogg}" = "xyes"; then
+    PKG_CHECK_MODULES(LIBVORBIS, [vorbis >= 1.1], [
+        AC_DEFINE(HAVE_LIBVORBIS, 1, [Define to 1 if you have the libvorbis])
+    ],[])
+fi
 PKG_ENABLE_MODULES_VLC([OGG], [], [ogg >= 1.0], [Ogg demux support], [auto], [${LIBVORBIS_CFLAGS}], [${LIBVORBIS_LIBS}])
 if test "${enable_sout}" != "no"; then
     PKG_ENABLE_MODULES_VLC([MUX_OGG], [], [ogg >= 1.0], [Ogg mux support], [auto])
-- 
1.9.0

