From ef4923d275024347497d4510c9a63e6e0f939b77 Mon Sep 17 00:00:00 2001
From: Ming Hu <tewilove@gmail.com>
Date: Mon, 23 Jun 2014 13:50:22 +0800
Subject: [PATCH] analysis: collect codec info for further purpose

---
 modules/demux/avformat/demux.c | 51 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 51 insertions(+)

diff --git a/modules/demux/avformat/demux.c b/modules/demux/avformat/demux.c
index bdd77fd..9ce8139 100644
--- a/modules/demux/avformat/demux.c
+++ b/modules/demux/avformat/demux.c
@@ -59,6 +59,10 @@
 
 # define HAVE_AVUTIL_CODEC_ATTACHMENT 1
 
+#include <jni.h>
+extern JavaVM *myVm;
+extern void jni_avformatSetCodecId(JNIEnv *, int, int);
+
 /*****************************************************************************
  * demux_sys_t: demux descriptor
  *****************************************************************************/
@@ -107,6 +111,47 @@ static vlc_fourcc_t CodecTagToFourcc( uint32_t codec_tag )
 #endif
 }
 
+static pthread_key_t g_thread_key;
+static pthread_once_t g_key_once = PTHREAD_ONCE_INIT;
+
+static void vlcjni_thread_destroy(void* value)
+{
+    JNIEnv *env = (JNIEnv*) value;
+    if (env != NULL) {
+        (*myVm)->DetachCurrentThread(myVm);
+        pthread_setspecific(g_thread_key, NULL);
+    }
+}
+
+static void vlcjni_make_thread_key()
+{
+    pthread_key_create(&g_thread_key, vlcjni_thread_destroy);
+}
+
+static jint vlcjni_setup_thread_env(JNIEnv **p_env)
+{
+    JavaVM *jvm = myVm;
+    if (!jvm)
+        return -1;
+
+    pthread_once(&g_key_once, vlcjni_make_thread_key);
+
+    JNIEnv *env = (JNIEnv*) pthread_getspecific(g_thread_key);
+    if (env) {
+        *p_env = env;
+        return 0;
+    }
+
+    static JavaVMAttachArgs vlc_thr_args = {JNI_VERSION_1_2, "AVFormat", NULL};
+    if ((*jvm)->AttachCurrentThread(jvm, &env, &vlc_thr_args) == JNI_OK) {
+        pthread_setspecific(g_thread_key, env);
+        *p_env = env;
+        return 0;
+    }
+
+    return -1;
+}
+
 /*****************************************************************************
  * Open
  *****************************************************************************/
@@ -343,6 +388,12 @@ int OpenDemux( vlc_object_t *p_this )
             continue;
 #endif
 
+        /* */
+        JNIEnv *env;
+
+        if (!vlcjni_setup_thread_env(&env))
+            jni_avformatSetCodecId(env, cc->codec_type, cc->codec_tag);
+
         switch( cc->codec_type )
         {
         case AVMEDIA_TYPE_AUDIO:
-- 
1.8.5.2 (Apple Git-48)

