From 47c41ffd353ebc86fc5d3859dc595c84caa56b70 Mon Sep 17 00:00:00 2001
From: Ming Hu <tewilove@gmail.com>
Date: Fri, 7 Mar 2014 11:15:21 +0800
Subject: [PATCH] ringbuf: fix read out of range in Peek()

---
 modules/stream_filter/ringbuf.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/stream_filter/ringbuf.c b/modules/stream_filter/ringbuf.c
index 7c5e26c..f9a36d2 100644
--- a/modules/stream_filter/ringbuf.c
+++ b/modules/stream_filter/ringbuf.c
@@ -366,6 +366,8 @@ static int PeekFromBuffer_l( stream_t *p_stream, void *p_buffer, int i_read )
     int i_to_read = i_data_ready;
     if( i_to_read > p_sys->i_buffer_size )
         i_to_read = p_sys->i_buffer_size;
+    if (i_to_read > i_read)
+        i_to_read = i_read;
 
     /* beyond last byte */
     int i_read_index  = p_sys->i_read_index;
-- 
1.8.5.3

