From 7fb984397df6ab69055e676a03ff15ca0067c30e Mon Sep 17 00:00:00 2001
From: Ming Hu <tewilove@gmail.com>
Date: Thu, 10 Jul 2014 12:41:05 +0800
Subject: [PATCH] mediacodec: check exception after queueInputBuffer

---
 modules/codec/omxil/android_mediacodec.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/codec/omxil/android_mediacodec.c b/modules/codec/omxil/android_mediacodec.c
index 041020f..aae1353 100644
--- a/modules/codec/omxil/android_mediacodec.c
+++ b/modules/codec/omxil/android_mediacodec.c
@@ -1128,6 +1128,12 @@ static picture_t *DecodeVideo(decoder_t *p_dec, block_t **pp_block)
         timestamp_FifoPut(p_sys->timestamp_fifo, p_block->i_pts ? VLC_TS_INVALID : p_block->i_dts);
         (*env)->CallVoidMethod(env, p_sys->codec, p_sys->queue_input_buffer, index, 0, size, ts, 0);
         (*env)->DeleteLocalRef(env, buf);
+        if ((*env)->ExceptionOccurred(env)) {
+            msg_Warn(p_dec, "Exception in MediaCodec.queueInputBuffer");
+            (*env)->ExceptionClear(env);
+            p_sys->error_state = true;
+            break;
+        }
         p_sys->decoded = true;
         break;
     }
-- 
2.0.0

