From 70f273c041eadf7d9b4286aaf548b6dc3f589cf4 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Fri, 6 Apr 2012 16:55:20 +0800
Subject: [PATCH] add shortcut for audiotrack module

---
 modules/audio_output/audiotrack.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/audio_output/audiotrack.c b/modules/audio_output/audiotrack.c
index 297c48e..2c3ef81 100644
--- a/modules/audio_output/audiotrack.c
+++ b/modules/audio_output/audiotrack.c
@@ -118,7 +118,7 @@ vlc_module_begin ()
     set_capability("audio output", 25)
     set_category(CAT_AUDIO)
     set_subcategory(SUBCAT_AUDIO_AOUT)
-    add_shortcut("android")
+    add_shortcut("androidaudiotrack", "android")
     set_callbacks(Open, Close)
 vlc_module_end ()
 
-- 
1.7.9.4

