From b6e632f2764af452042e6600037a06e82e487695 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Fri, 23 Mar 2012 19:29:10 +0800
Subject: [PATCH] disable live555 and rtsp

---
 extras/package/android/configure.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extras/package/android/configure.sh b/extras/package/android/configure.sh
index 4af9cc2..5c6b9ff 100755
--- a/extras/package/android/configure.sh
+++ b/extras/package/android/configure.sh
@@ -41,7 +41,7 @@ STRIP="${CROSS_COMPILE}strip" \
 RANLIB="${CROSS_COMPILE}ranlib" \
 AR="${CROSS_COMPILE}ar" \
 sh $VLC_SOURCEDIR/configure --host=arm-linux-androideabi --build=x86_64-unknown-linux $EXTRA_PARAMS \
-                --enable-live555 --enable-realrtsp \
+                --disable-live555 --disable-realrtsp \
                 --enable-avformat \
                 --enable-swscale \
                 --enable-avcodec \
-- 
1.7.9.4

