From 0107a5f8da9ed74139b8d36045b158a8b84b236b Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 26 Mar 2012 20:36:56 +0800
Subject: [PATCH] do not add range header if position is zero

---
 modules/access/http.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/modules/access/http.c b/modules/access/http.c
index a292ebb..ffa9faf 100644
--- a/modules/access/http.c
+++ b/modules/access/http.c
@@ -1292,8 +1292,11 @@ static int Request( access_t *p_access, uint64_t i_tell )
     if( p_sys->i_version == 1 && ! p_sys->b_continuous )
     {
         p_sys->b_persist = true;
-        net_Printf( p_access, p_sys->fd, pvs,
+        if (i_tell != 0)
+        {
+            net_Printf( p_access, p_sys->fd, pvs,
                     "Range: bytes=%"PRIu64"-\r\n", i_tell );
+        }
         net_Printf( p_access, p_sys->fd, pvs, "Connection: close\r\n" );
     }
 
-- 
1.7.9.4

