From 7d4442e2fedd09fa527acaee38ca98bcb04123dc Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 27 Mar 2012 15:31:08 +0800
Subject: [PATCH] ts demux need fetch first pcr for video time

---
 modules/demux/ts.c |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/modules/demux/ts.c b/modules/demux/ts.c
index f9c2f4d..022df6a 100644
--- a/modules/demux/ts.c
+++ b/modules/demux/ts.c
@@ -695,6 +695,16 @@ static int Open( vlc_object_t *p_this )
         CheckPCR( p_demux );
         GetLastPCR( p_demux );
     }
+    else
+    {
+        /* obtain p_sys->i_first_pcr if possible. DEMUX_GET_TIME need it */
+        bool can_normal_seek = false;
+        stream_Control( p_demux->s, STREAM_CAN_SEEK, &can_normal_seek );
+        if( can_normal_seek )
+        {
+            GetFirstPCR( p_demux );            
+        }
+    }
     if( p_sys->i_first_pcr < 0 || p_sys->i_last_pcr < 0 )
     {
         p_sys->b_force_seek_per_percent = true;
@@ -991,6 +1001,12 @@ static int Control( demux_t *p_demux, int i_query, va_list args )
             {
                 *pi64 = 0;
             }
+            
+            if( *pi64 == 0 && p_sys->i_first_pcr >= 0 &&
+                p_sys->i_first_pcr < p_sys->i_current_pcr)
+            {
+                *pi64 = (p_sys->i_current_pcr - p_sys->i_first_pcr) * 100 / 9;
+            }
         }
         else
         {
-- 
1.7.9.4

