From 7506b82ec75e81471f2b4cca4ff38f210a95054e Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 9 Apr 2012 20:00:23 +0800
Subject: [PATCH] Fix the msleep nanosleep wrapping

---
 src/posix/thread.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/posix/thread.c b/src/posix/thread.c
index 598c692..6c36980 100644
--- a/src/posix/thread.c
+++ b/src/posix/thread.c
@@ -1156,7 +1156,7 @@ void msleep (mtime_t delay)
     vlc_testcancel();
     for (;;) {
         struct timespec t = { 0, 10 * 1000 * 1000 };
-        if (t.tv_nsec < ts.tv_nsec)
+        if (ts.tv_sec <= 0 && t.tv_nsec > ts.tv_nsec)
             t.tv_nsec = ts.tv_nsec;
         while (nanosleep (&t, &t) == -1) {
             vlc_testcancel();
-- 
1.7.9.4

