From dfd10b6faf599b48ae31a294ad1286bac5fd97dc Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Fri, 23 Mar 2012 17:59:42 +0800
Subject: [PATCH] lite-mode: Android.mk: disable modules

---
 vlc-android/jni/Android.mk | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/vlc-android/jni/Android.mk b/vlc-android/jni/Android.mk
index 8373aa8..a9a12da 100644
--- a/vlc-android/jni/Android.mk
+++ b/vlc-android/jni/Android.mk
@@ -69,17 +69,12 @@ LOCAL_LDLIBS := -L$(VLC_CONTRIB)/lib \
 	$(VLC_BUILD_DIR)/compat/.libs/libcompat.a \
 	-ldl -lz -lm -llog \
 	-landroid \
-	-ldvbpsi -lebml -lmatroska -ltag \
-	-logg -lFLAC -ltheora -lvorbis \
-	-lmpeg2 -la52 \
-	-lavformat -lavcodec -lswscale -lavutil -lpostproc -lgsm -lopenjpeg \
-	-lliveMedia -lUsageEnvironment -lBasicUsageEnvironment -lgroupsock \
-	-lspeex -lspeexdsp \
-	-lxml2 -lpng -lgnutls -lgcrypt -lgpg-error \
+	-ldvbpsi \
+	-lavformat -lavcodec -lswscale -lavutil \
+	-lgnutls -lgcrypt -lgpg-error \
 	-lnettle -lhogweed -lgmp \
-	-lfreetype -liconv -lass -lfribidi -lopus \
-	-lEGL -lGLESv2 -ljpeg \
-	-ldvdnav -ldvdread -ldvdcss \
+	-lEGL -lGLESv2 \
+	-lfaad \
 	$(CPP_STATIC)
 
 include $(BUILD_SHARED_LIBRARY)
-- 
1.8.5.2 (Apple Git-48)

