From 073c01d9e125873f21deccbb2501b9cb6f60fb96 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 28 May 2012 11:45:12 +0800
Subject: [PATCH 2/6] lite-mode: compile.sh: disable contrib

---
 compile.sh | 39 ++++++++++++++++++++++++++++++++++-----
 1 file changed, 34 insertions(+), 5 deletions(-)

diff --git a/compile.sh b/compile.sh
index aa3dbdf..b911306 100755
--- a/compile.sh
+++ b/compile.sh
@@ -215,8 +215,8 @@ gen_pc_file GLESv2 2
 
 cd contrib/android
 ../bootstrap --host=${TARGET_TUPLE} --disable-disc --disable-sout \
-    --enable-dvdread \
-    --enable-dvdnav \
+    --disable-dvdread \
+    --disable-dvdnav \
     --disable-dca \
     --disable-goom \
     --disable-chromaprint \
@@ -233,12 +233,41 @@ cd contrib/android
     --disable-upnp \
     --disable-gme \
     --disable-tremor \
-    --enable-vorbis \
+    --disable-vorbis \
     --disable-sidplay2 \
+    --disable-ass \
+    --disable-fribidi \
+    --disable-taglib \
+    --disable-live555 \
+    --disable-schroedinger \
+    --disable-matroska \
+    --disable-goom \
+    --disable-gsm \
+    --disable-speex \
+    --disable-speexdsp \
+    --disable-freetype2 \
+    --disable-lua \
+    --disable-flac \
+    --disable-ogg \
+    --disable-dca \
+    --disable-theora \
+    --disable-modplug \
+    --disable-png \
+    --disable-openjpg \
+    --disable-libxml2 \
+    --disable-opus \
+    --disable-mad \
+    --disable-postproc \
+    --disable-libmpeg2 \
+    --disable-a52 \
+    --disable-vncserver \
+    --disable-png \
+    --disable-jpeg \
     --disable-samplerate \
-    --disable-faad2 \
+    --enable-faad2 \
     --disable-harfbuzz \
-    --enable-iconv
+    --disable-ssh2 \
+    --disable-iconv
 
 # TODO: mpeg2, theora
 
-- 
2.0.0

