From a42903908b81d264aea9daf7f7859bf9f3267cac Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 28 May 2012 12:04:30 +0800
Subject: [PATCH 4/4] lite-mode: find_modules.sh: blacklist modules

---
 find_modules.sh | 39 ++++++++++++++++++++++++++++++++++++++-
 1 file changed, 38 insertions(+), 1 deletion(-)

diff --git a/find_modules.sh b/find_modules.sh
index 6e70bb8..0be2ff5 100755
--- a/find_modules.sh
+++ b/find_modules.sh
@@ -52,12 +52,49 @@ magnify
 gradient
 .*tospdif
 dtstofloat32
-logger
 visual
 fb
+aiff
 aout_file
 yuv
 .dummy
+a52*
+access_(attachment|mms|rar|udp|vrd)
+aout_file
+asf
+avi
+ay
+compressor
+decomp
+dirac
+dts
+flac
+flacsys
+folder
+karaoke
+marq
+libmpeg2
+ogg
+osd_parser
+packetizer_(dirac|flac)
+postproc
+playlist
+rtp
+sdp
+speex
+stream_filter_(dash|rar)
+subtitle
+(cvd|svc|dvd)sub
+subsde
+subsdelay
+tta
+visual
+vobsub
+voc
+wav
+wave
+xml
+zip
 "
 
 regexp=
-- 
1.9.0

