From e77efd19c8ce0c7918fac666da32d110c86c75a2 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 16 Apr 2012 23:20:57 +0800
Subject: [PATCH 3/5] lite-mode: configure.sh: disable modules

---
 configure.sh |   10 ++++++----
 1 个文件被修改，插入 6 行(+)，删除 4 行(-)

diff --git a/configure.sh b/configure.sh
index 6546e27..b915890 100755
--- a/configure.sh
+++ b/configure.sh
@@ -40,15 +40,17 @@ STRIP="${CROSS_COMPILE}strip" \
 RANLIB="${CROSS_COMPILE}ranlib" \
 AR="${CROSS_COMPILE}ar" \
 sh $VLC_SOURCEDIR/configure --host=arm-linux-androideabi --build=x86_64-unknown-linux $EXTRA_PARAMS \
-                --enable-live555 --enable-realrtsp \
+                --disable-live555 --disable-realrtsp \
                 --enable-avformat \
                 --enable-swscale \
                 --enable-avcodec \
                 --enable-opensles \
                 --enable-audiotrack \
                 --enable-android-surface \
-                --enable-mkv \
-                --enable-taglib \
+                --enable-debug \
+                --disable-mkv \
+                --disable-taglib \
+                --disable-freetype \
                 --enable-iomx \
                 --disable-vlc --disable-shared \
                 --disable-vlm \
@@ -84,7 +86,7 @@ sh $VLC_SOURCEDIR/configure --host=arm-linux-androideabi --build=x86_64-unknown-
                 --disable-skins2 \
                 --disable-mtp \
                 --disable-notify \
-                --enable-libass \
+                --disable-libass \
                 --disable-svg \
                 --disable-sqlite \
                 --disable-udev \
-- 
1.7.10.4

