From 1c1e9e06e781b35355f50d75197534e59c0e2f30 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 16 Apr 2012 23:20:57 +0800
Subject: [PATCH 3/5] lite-mode: configure.sh: disable modules

---
 configure.sh |    9 +++++----
 1 个文件被修改，插入 5 行(+)，删除 4 行(-)

diff --git a/configure.sh b/configure.sh
index e5a3c71..48c3e79 100755
--- a/configure.sh
+++ b/configure.sh
@@ -46,7 +46,7 @@ STRIP="${CROSS_COMPILE}strip" \
 RANLIB="${CROSS_COMPILE}ranlib" \
 AR="${CROSS_COMPILE}ar" \
 sh $VLC_SOURCEDIR/configure --host=arm-linux-androideabi --build=x86_64-unknown-linux $EXTRA_PARAMS \
-                --enable-live555 --enable-realrtsp \
+                --disable-live555 --disable-realrtsp \
                 --enable-avformat \
                 --enable-swscale \
                 --enable-avcodec \
@@ -54,8 +54,9 @@ sh $VLC_SOURCEDIR/configure --host=arm-linux-androideabi --build=x86_64-unknown-
                 --enable-audiotrack \
                 --enable-android-surface \
                 --enable-debug \
-                --enable-mkv \
-                --enable-taglib \
+                --disable-mkv \
+                --disable-taglib \
+                --disable-freetype \
                 --enable-iomx \
                 --disable-vlc --disable-shared \
                 --disable-vlm \
@@ -91,7 +92,7 @@ sh $VLC_SOURCEDIR/configure --host=arm-linux-androideabi --build=x86_64-unknown-
                 --disable-skins2 \
                 --disable-mtp \
                 --disable-notify \
-                --enable-libass \
+                --disable-libass \
                 --disable-svg \
                 --disable-sqlite \
                 --disable-udev \
-- 
1.7.10.2

