From 15814b29e70ded0afad2327e736c9d6b7b2badca Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 24 Jun 2012 17:54:36 +0800
Subject: [PATCH] build for armv5

---
 compile.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/compile.sh b/compile.sh
index 4024d88..b1f3e7c 100755
--- a/compile.sh
+++ b/compile.sh
@@ -1,5 +1,11 @@
 #! /bin/sh
 
+# patched for armv5
+export ANDROID_ABI=armeabi
+export NO_NEON=1
+export NO_FPU=1
+export NO_ARMV6=1
+
 # Read the Android Wiki http://wiki.videolan.org/AndroidCompile
 # Setup all that stuff correctly.
 # Get the latest Android SDK Platform or modify numbers in configure.sh and vlc-android/default.properties.
-- 
1.8.3.4

