From 94eb285e38d2dd0eb43a65ecaf53aaaf1105c791 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 24 Jun 2012 17:23:34 +0800
Subject: [PATCH] build for x86

---
 compile.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/compile.sh b/compile.sh
index 4024d88..785ea2e 100755
--- a/compile.sh
+++ b/compile.sh
@@ -1,5 +1,8 @@
 #! /bin/sh
 
+# patched for x86
+export ANDROID_ABI=x86
+
 # Read the Android Wiki http://wiki.videolan.org/AndroidCompile
 # Setup all that stuff correctly.
 # Get the latest Android SDK Platform or modify numbers in configure.sh and vlc-android/default.properties.
-- 
1.8.3.4

