From 345f230b2b1474bfef62fd99bb6db03190de975b Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Fri, 23 Mar 2012 17:59:42 +0800
Subject: [PATCH 1/4] lite-mode: Android.mk: disable modules

---
 vlc-android/jni/Android.mk | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/vlc-android/jni/Android.mk b/vlc-android/jni/Android.mk
index 0b6d115..adc2d7b 100644
--- a/vlc-android/jni/Android.mk
+++ b/vlc-android/jni/Android.mk
@@ -63,15 +63,11 @@ LOCAL_LDLIBS := -L$(VLC_CONTRIB)/lib \
 	$(VLC_BUILD_DIR)/compat/.libs/libcompat.a \
 	-ldl -lz -lm -llog \
 	-landroid \
-	-ldvbpsi -lebml -lmatroska -ltag \
-	-logg -lFLAC -ltheora \
-	-lmpeg2 -la52 \
-	-lavformat -lavcodec -lswscale -lavutil -lpostproc -lgsm -lopenjpeg \
-	-lliveMedia -lUsageEnvironment -lBasicUsageEnvironment -lgroupsock \
-	-lspeex -lspeexdsp \
-	-lxml2 -lpng -lgnutls -lgcrypt -lgpg-error \
+	-ldvbpsi \
+	-lavformat -lavcodec -lswscale -lavutil \
+	-lgnutls -lgcrypt -lgpg-error \
 	-lnettle -lhogweed -lgmp \
-	-lfreetype -liconv -lass -lfribidi -lopus \
+	-lfaad \
 	$(CPP_STATIC)
 
 include $(BUILD_SHARED_LIBRARY)
-- 
1.8.3.4

