From d362a309d0d4d37b593bc5e45e63a49fc147b254 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Fri, 23 Mar 2012 17:59:42 +0800
Subject: [PATCH 1/4] lite-mode: Android.mk: disable modules

---
 vlc-android/jni/Android.mk | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/vlc-android/jni/Android.mk b/vlc-android/jni/Android.mk
index cc104b7..a9a12da 100644
--- a/vlc-android/jni/Android.mk
+++ b/vlc-android/jni/Android.mk
@@ -69,16 +69,12 @@ LOCAL_LDLIBS := -L$(VLC_CONTRIB)/lib \
 	$(VLC_BUILD_DIR)/compat/.libs/libcompat.a \
 	-ldl -lz -lm -llog \
 	-landroid \
-	-ldvbpsi -lebml -lmatroska -ltag \
-	-logg -lFLAC -ltheora \
-	-lmpeg2 -la52 \
-	-lavformat -lavcodec -lswscale -lavutil -lpostproc -lgsm -lopenjpeg \
-	-lliveMedia -lUsageEnvironment -lBasicUsageEnvironment -lgroupsock \
-	-lspeex -lspeexdsp \
-	-lxml2 -lpng -lgnutls -lgcrypt -lgpg-error \
+	-ldvbpsi \
+	-lavformat -lavcodec -lswscale -lavutil \
+	-lgnutls -lgcrypt -lgpg-error \
 	-lnettle -lhogweed -lgmp \
-	-lfreetype -liconv -lass -lfribidi -lopus \
-	-lEGL -lGLESv2 -ljpeg \
+	-lEGL -lGLESv2 \
+	-lfaad \
 	$(CPP_STATIC)
 
 include $(BUILD_SHARED_LIBRARY)
-- 
1.8.3.4

