From 6deff7c173f03d4dda7e4d6d2f4c96291bcaa679 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Fri, 23 Mar 2012 17:59:42 +0800
Subject: [PATCH] lite-mode: Android.mk: disable modules

---
 vlc-android/jni/Android.mk |   10 +++-------
 1 个文件被修改，插入 3 行(+)，删除 7 行(-)

diff --git a/vlc-android/jni/Android.mk b/vlc-android/jni/Android.mk
index 1cf902d..2f8661a 100644
--- a/vlc-android/jni/Android.mk
+++ b/vlc-android/jni/Android.mk
@@ -60,14 +60,10 @@ LOCAL_LDLIBS := -L$(VLC_CONTRIB)/lib \
 	$(VLC_BUILD_DIR)/src/.libs/libvlccore.a \
 	$(VLC_BUILD_DIR)/compat/.libs/libcompat.a \
 	-ldl -lz -lm -llog \
-	-ldvbpsi -lebml -lmatroska -ltag \
-	-logg -lFLAC -ltheora \
+	-ldvbpsi \
 	-lmpeg2 -la52 \
-	-lavformat -lavcodec -lswscale -lavutil -lpostproc -lgsm -lopenjpeg \
-	-lliveMedia -lUsageEnvironment -lBasicUsageEnvironment -lgroupsock \
-	-lspeex -lspeexdsp \
-	-lxml2 -lpng -lgnutls -lgcrypt -lgpg-error \
-	-lfreetype -liconv -lass -lfribidi -lopus \
+	-lavformat -lavcodec -lswscale -lavutil -lpostproc \
+	-lgnutls -lgcrypt -lgpg-error \
 	$(CPP_STATIC)
 
 include $(BUILD_SHARED_LIBRARY)
-- 
1.7.10.4

