From a43a7d11bc8945726e6d94ebcb899bb88bd314df Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 28 May 2012 11:45:12 +0800
Subject: [PATCH 2/4] lite-mode: compile.sh: disable contrib

---
 compile.sh | 35 ++++++++++++++++++++++++++++++++---
 1 file changed, 32 insertions(+), 3 deletions(-)

diff --git a/compile.sh b/compile.sh
index a29cd55..e614b4a 100755
--- a/compile.sh
+++ b/compile.sh
@@ -231,12 +231,41 @@ cd contrib/android
     --disable-upnp \
     --disable-gme \
     --disable-tremor \
-    --enable-vorbis \
+    --disable-vorbis \
     --disable-sidplay2 \
+    --disable-ass \
+    --disable-fribidi \
+    --disable-taglib \
+    --disable-live555 \
+    --disable-schroedinger \
+    --disable-matroska \
+    --disable-goom \
+    --disable-gsm \
+    --disable-speex \
+    --disable-speexdsp \
+    --disable-freetype2 \
+    --disable-lua \
+    --disable-flac \
+    --disable-ogg \
+    --disable-dca \
+    --disable-theora \
+    --disable-modplug \
+    --disable-png \
+    --disable-openjpg \
+    --disable-libxml2 \
+    --disable-opus \
+    --disable-mad \
+    --disable-postproc \
+    --disable-libmpeg2 \
+    --disable-a52 \
+    --disable-vncserver \
+    --disable-png \
+    --disable-jpeg \
     --disable-samplerate \
-    --disable-faad2 \
+    --enable-faad2 \
     --disable-harfbuzz \
-    --enable-iconv
+    --disable-ssh2 \
+    --disable-iconv
 
 # TODO: mpeg2, theora
 
-- 
1.8.3.4

