From 112cca08dbc2c4b300ecec1e49ff20ca52eacfa5 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 16 Apr 2012 23:20:57 +0800
Subject: [PATCH 3/4] lite-mode: configure.sh: disable modules

---
 configure.sh | 22 ++++++++++++++--------
 1 file changed, 14 insertions(+), 8 deletions(-)

diff --git a/configure.sh b/configure.sh
index c8643b9..0936599 100755
--- a/configure.sh
+++ b/configure.sh
@@ -43,17 +43,20 @@ NM="${CROSS_COMPILE}nm" \
 STRIP="${CROSS_COMPILE}strip" \
 RANLIB="${CROSS_COMPILE}ranlib" \
 AR="${CROSS_COMPILE}ar" \
-sh $VLC_SOURCEDIR/configure --host=$TARGET_TUPLE --build=x86_64-unknown-linux $EXTRA_PARAMS \
-                --enable-live555 --enable-realrtsp \
+MODULE_LITE=1 \
+sh $VLC_SOURCEDIR/configure --host=arm-linux-androideabi --build=x86_64-unknown-linux $EXTRA_PARAMS \
+                --disable-live555 --disable-realrtsp \
                 --enable-avformat \
                 --enable-swscale \
                 --enable-avcodec \
-                --enable-opus \
+                --disable-opus \
                 --enable-opensles \
                 --enable-android-surface \
-                --enable-mkv \
-                --enable-taglib \
+                --enable-debug \
+                --disable-mkv \
+                --disable-taglib \
                 --enable-dvbpsi \
+                --disable-freetype \
                 --disable-vlc --disable-shared \
                 --disable-update-check \
                 --disable-vlm \
@@ -89,10 +92,10 @@ sh $VLC_SOURCEDIR/configure --host=$TARGET_TUPLE --build=x86_64-unknown-linux $E
                 --disable-skins2 \
                 --disable-mtp \
                 --disable-notify \
-                --enable-libass \
+                --disable-libass \
                 --disable-svg \
                 --disable-udev \
-                --enable-libxml2 \
+                --disable-libxml2 \
                 --disable-caca \
                 --disable-glx \
                 --disable-egl \
@@ -100,7 +103,10 @@ sh $VLC_SOURCEDIR/configure --host=$TARGET_TUPLE --build=x86_64-unknown-linux $E
                 --disable-projectm \
                 --disable-sout \
                 --disable-vorbis \
-                --disable-faad \
+                --enable-faad \
                 --disable-x264 \
+                --disable-postproc \
+                --disable-a52 \
+                --disable-libmpeg2 \
                 --disable-schroedinger --disable-dirac \
                 $*
-- 
1.8.2.1

