From 90d08de9eb0c662eb46016c2e313927478f9507e Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 16 Apr 2012 23:20:57 +0800
Subject: [PATCH 3/4] lite-mode: configure.sh: disable modules

---
 configure.sh |   16 +++++++++-------
 1 个文件被修改，插入 9 行(+)，删除 7 行(-)

diff --git a/configure.sh b/configure.sh
index 2383cf7..538aae0 100755
--- a/configure.sh
+++ b/configure.sh
@@ -49,18 +49,20 @@ NM="${CROSS_COMPILE}nm" \
 STRIP="${CROSS_COMPILE}strip" \
 RANLIB="${CROSS_COMPILE}ranlib" \
 AR="${CROSS_COMPILE}ar" \
-sh $VLC_SOURCEDIR/configure --host=$TARGET_TUPLE --build=x86_64-unknown-linux $EXTRA_PARAMS \
-                --enable-live555 --enable-realrtsp \
+sh $VLC_SOURCEDIR/configure --host=arm-linux-androideabi --build=x86_64-unknown-linux $EXTRA_PARAMS \
+                --disable-live555 --disable-realrtsp \
                 --enable-avformat \
                 --enable-swscale \
                 --enable-avcodec \
-                --enable-opus \
+                --disable-opus \
                 --enable-opensles \
                 --enable-audiotrack \
                 --enable-android-surface \
-                --enable-mkv \
-                --enable-taglib \
+                --enable-debug \
+                --disable-mkv \
+                --disable-taglib \
                 --enable-dvbpsi \
+                --disable-freetype \
                 --enable-iomx \
                 --disable-vlc --disable-shared \
                 --disable-vlm \
@@ -96,11 +98,11 @@ sh $VLC_SOURCEDIR/configure --host=$TARGET_TUPLE --build=x86_64-unknown-linux $E
                 --disable-skins2 \
                 --disable-mtp \
                 --disable-notify \
-                --enable-libass \
+                --disable-libass \
                 --disable-svg \
                 --disable-sqlite \
                 --disable-udev \
-                --enable-libxml2 \
+                --disable-libxml2 \
                 --disable-caca \
                 --disable-glx \
                 --disable-egl \
-- 
1.7.10.4

