From 4c4efafc39a375835bb408d3e479b8b9d9298feb Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 28 May 2012 12:04:30 +0800
Subject: [PATCH 4/4] lite-mode: find_modules.sh: blacklist modules

---
 find_modules.sh |   36 +++++++++++++++++++++++++++++++++++-
 1 个文件被修改，插入 35 行(+)，删除 1 行(-)

diff --git a/find_modules.sh b/find_modules.sh
index 322033a..7b2eab4 100755
--- a/find_modules.sh
+++ b/find_modules.sh
@@ -51,12 +51,46 @@ magnify
 gradient
 .*tospdif
 dtstofloat32
-logger
 visual
 fb
+aiff
 aout_file
 yuv
 .dummy
+access_(attachment|mms|rar|udp|vrd)
+aout_file
+asf
+avi
+ay
+compressor
+decomp
+dirac
+dts
+flac
+flacsys
+folder
+karaoke
+marq
+ogg
+osd_parser
+packetizer_(dirac|flac)
+playlist
+rtp
+sdp
+speex
+stream_filter_(dash|rar)
+subtitle
+(cvd|svc|dvd)sub
+subsde
+subsdelay
+tta
+visual
+vobsub
+voc
+wav
+wave
+xml
+zip
 "
 
 regexp=
-- 
1.7.10.4

