From 536458384bda584a18d055a44009ca53c860761e Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 24 Jun 2012 17:54:36 +0800
Subject: [PATCH] build for armv6-vfp

---
 compile.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/compile.sh b/compile.sh
index d025c2e..cdfb702 100755
--- a/compile.sh
+++ b/compile.sh
@@ -6,6 +6,10 @@
 
 set -e
 
+# patched for armv6-vfp
+export ANDROID_ABI=armeabi
+export NO_NEON=1
+
 if [ -z "$ANDROID_NDK" -o -z "$ANDROID_SDK" ]; then
    echo "You must define ANDROID_NDK, ANDROID_SDK and ANDROID_ABI before starting."
    echo "They must point to your NDK and SDK directories.\n"
-- 
1.8.3.4

