From 3193db5574eef55cb2051e73b550c02c12829842 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 24 Jun 2012 17:54:36 +0800
Subject: [PATCH] build for armv6-vfp

---
 compile.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/compile.sh b/compile.sh
index 4024d88..45685f4 100755
--- a/compile.sh
+++ b/compile.sh
@@ -1,11 +1,16 @@
 #! /bin/sh
 
+# patched for armv6-vfp
+export ANDROID_ABI=armeabi
+export NO_NEON=1
+
 # Read the Android Wiki http://wiki.videolan.org/AndroidCompile
 # Setup all that stuff correctly.
 # Get the latest Android SDK Platform or modify numbers in configure.sh and vlc-android/default.properties.
 
 set -e
 
+<<<<<<< HEAD
 BUILD=
 FETCH=
 case "$1" in
-- 
1.8.3.4

