From 4725b275af2eb0bb206e2589136fdbb96bac7e4c Mon Sep 17 00:00:00 2001
From: Ming Hu <tewilove@gmail.com>
Date: Mon, 9 Jun 2014 10:47:27 +0800
Subject: [PATCH 1003/1003] contrib: enable minium remuxing through ffmpeg

---
 contrib/src/ffmpeg/rules.mak | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 321a491..931d07c 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -102,6 +102,11 @@ FFMEPGCONF += \
 	--enable-decoder=vc1_vdpau \
 	--enable-decoder=vc1image
 
+# Support remuxing to certain formats
+FFMPEGCONF += --enable-protocol=file
+FFMPEGCONF += --enable-muxer=mp4
+FFMPEGCONF += --enable-bsf=aac_adtstoasc
+
 ifdef HAVE_CROSS_COMPILE
 FFMPEGCONF += --enable-cross-compile
 ifndef HAVE_DARWIN_OS
-- 
1.8.5.2 (Apple Git-48)

