From e0b57a13bf07e8e0b815327506e548037f286782 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 29 Mar 2012 16:55:54 +0800
Subject: [PATCH] build-for-armv6j

---
 compile.sh |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/compile.sh b/compile.sh
index d172500..20ced80 100755
--- a/compile.sh
+++ b/compile.sh
@@ -19,6 +19,11 @@ export NO_NDK_V7=0
 
 EOF
 
+
+# patched for armv6j
+export NO_NEON=1
+
+
 if [ -z "$ANDROID_NDK" -o -z "$ANDROID_SDK" ]; then
    echo "You must define ANDROID_NDK and ANDROID_SDK before starting."
    echo "They must point to your NDK and SDK directories."
-- 
1.7.9.4

