From 32e8579e6652c5cf6c6a6ba702891bb6885a444a Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 24 Jun 2012 17:54:36 +0800
Subject: [PATCH] build for armv6

---
 compile.sh |    5 +++++
 1 个文件被修改，插入 5 行(+)

diff --git a/compile.sh b/compile.sh
index 4add5ae..0357404 100755
--- a/compile.sh
+++ b/compile.sh
@@ -6,6 +6,11 @@
 
 set -e
 
+# patched for armv6
+export ANDROID_ABI=armeabi
+export NO_NEON=1
+export NO_FPU=1
+
 if [ -z "$ANDROID_NDK" -o -z "$ANDROID_SDK" -o -z "$ANDROID_ABI" ]; then
    echo "You must define ANDROID_NDK, ANDROID_SDK and ANDROID_ABI before starting."
    echo "They must point to your NDK and SDK directories.\n"
-- 
1.7.10.4

