From 078265315c8b2d66d43e2e48bae6e900b4b4ba3c Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 24 Jun 2012 17:54:36 +0800
Subject: [PATCH] build for armv6

---
 compile.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/compile.sh b/compile.sh
index d025c2e..3fdb07d 100755
--- a/compile.sh
+++ b/compile.sh
@@ -6,6 +6,11 @@
 
 set -e
 
+# patched for armv6
+export ANDROID_ABI=armeabi
+export NO_NEON=1
+export NO_FPU=1
+
 if [ -z "$ANDROID_NDK" -o -z "$ANDROID_SDK" ]; then
    echo "You must define ANDROID_NDK, ANDROID_SDK and ANDROID_ABI before starting."
    echo "They must point to your NDK and SDK directories.\n"
-- 
1.8.3.4

