From 35d80c1c549ae833acd9e8d84a57ed6d888fe031 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 24 Jun 2012 17:54:36 +0800
Subject: [PATCH] build for armv6

---
 compile.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/compile.sh b/compile.sh
index 4024d88..3765b67 100755
--- a/compile.sh
+++ b/compile.sh
@@ -1,5 +1,10 @@
 #! /bin/sh
 
+# patched for armv6
+export ANDROID_ABI=armeabi
+export NO_NEON=1
+export NO_FPU=1
+
 # Read the Android Wiki http://wiki.videolan.org/AndroidCompile
 # Setup all that stuff correctly.
 # Get the latest Android SDK Platform or modify numbers in configure.sh and vlc-android/default.properties.
-- 
1.8.3.4

