From c9976db1bdba9b91acd95cb4f0b8825cb1f4e8be Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 18:21:02 +0800
Subject: [PATCH] fix toolchain config

---
 contrib/src/main.mak | 1 +
 1 file changed, 1 insertion(+)

diff --git a/contrib/src/main.mak b/contrib/src/main.mak
index 69465e6..811ad19 100644
--- a/contrib/src/main.mak
+++ b/contrib/src/main.mak
@@ -397,6 +397,7 @@ ifdef HAVE_CROSS_COMPILE
 endif
 	echo "set(CMAKE_C_COMPILER $(CC))" >> $@
 	echo "set(CMAKE_CXX_COMPILER $(CXX))" >> $@
+	echo "set(_CMAKE_TOOLCHAIN_PREFIX $(ANDROID_NDK)/toolchains/arm-linux-androideabi-4.7/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-)" >> $@
 	echo "set(CMAKE_FIND_ROOT_PATH $(PREFIX))" >> $@
 	echo "set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)" >> $@
 	echo "set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)" >> $@
-- 
1.8.2.1

