From 271648180e883f0c8d1efde632e945fd84677a36 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 17:18:45 +0800
Subject: [PATCH 1/5] modify compile.sh for MacOS build

---
 compile.sh |   30 ++++++++++++++++++++++++++++++
 1 个文件被修改，插入 30 行(+)

diff --git a/compile.sh b/compile.sh
index a971ac6..9330aa2 100755
--- a/compile.sh
+++ b/compile.sh
@@ -43,6 +43,17 @@ fi
 
 # Add the NDK toolchain to the PATH, needed both for contribs and for building
 # stub libraries
+BUILD_PLATFORM='unknown'
+if [ "$(uname)" == 'Linux' ]; then
+   BUILD_PLATFORM='linux'
+elif [ "$(uname)" == 'Darwin' ]; then
+   BUILD_PLATFORM='darwin'
+fi
+
+if [ ${BUILD_PLATFORM} == 'darwin' ]; then
+    export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/darwin-x86/bin:${PATH}
+fi
+
 export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/linux-x86/bin:${PATH}
 
 # 1/ libvlc, libvlccore and its plugins
@@ -54,6 +65,14 @@ if [ ! -d "vlc" ]; then
     git checkout -B android ${TESTED_HASH}
     echo "Applying the patches"
     git am ../patches/*.patch
+
+    if [ ${BUILD_PLATFORM} == 'darwin' ]; then
+        git am ../../patches/vlc-on-macosx/*.patch
+    fi
+    git am ../../patches/vlc/*.patch
+elif [ '$1' != '-f' ]; then
+    echo "VLC source found, build without hash check"
+    cd vlc
 else
     echo "VLC source found"
     cd vlc
@@ -70,6 +89,10 @@ git reset --hard origin
 git pull origin master
 git checkout -B android ${TESTED_HASH}
 git am ../patches/*
+if [[ ${BUILD_PLATFORM} == 'darwin' ]]; then
+    git am ../../patches/vlc-on-macosx/*.patch
+fi
+git am ../../patches/vlc/*.patch
 
 *** : This will delete any changes you made to the current branch ***
 
@@ -78,6 +101,7 @@ EOF
     fi
 fi
 
+
 echo "Building the contribs"
 mkdir -p contrib/android
 cd contrib/android
@@ -108,6 +132,12 @@ if test -z "${NO_NEON}" -o -n "${TEGRA2}"; then
     echo "NOTHUMB := -marm" >> config.mak
 fi
 
+if  test -d "../../../../tarballs" ; then
+    echo "fetch local tarballs"
+    mkdir -p ../tarballs
+    cp -n -v ../../../../tarballs/* ../tarballs/ || true
+fi
+
 make fetch
 make
 
-- 
1.7.10.2

