From 3c55a4cf713bc02d0b6018937d8f50453b150e5c Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 14 Jul 2012 15:15:42 +0800
Subject: [PATCH 1/6] .gitignore

---
 .gitignore |    2 ++
 1 个文件被修改，插入 2 行(+)

diff --git a/.gitignore b/.gitignore
index b5ae6d4..f3c14ee 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,7 @@
 /vlc/
 /android-libs/*.so
+/patches/100*.patch
+/fastbuild.sh
 vlc-android/local.properties
 .DS_Store
 *.keystore
-- 
1.7.10.4

