From ff8517c0bc83fd9b3ae4f62f8993b7e5960c3d39 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 14 Jul 2012 15:15:42 +0800
Subject: [PATCH 01/12] .gitignore

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 32a12cc..93a51fb 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,6 @@
 /vlc/
 /android-libs/*.so
+/patches/100*.patch
 local.properties
 .DS_Store
 *.keystore
-- 
1.8.3.4

