From fc7b0e8faa1fcc29188d2b676abb5975ce1c709f Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 17:18:45 +0800
Subject: [PATCH 1/8] modify compile.sh for MacOS build

---
 compile.sh |   21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/compile.sh b/compile.sh
index 8c2d614..0d689fb 100755
--- a/compile.sh
+++ b/compile.sh
@@ -37,7 +37,7 @@ fi
 
 # Add the NDK toolchain to the PATH, needed both for contribs and for building
 # stub libraries
-export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/linux-x86/bin:${PATH}
+export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/darwin-x86/bin:${PATH}
 
 # 1/ libvlc, libvlccore and its plugins
 TESTED_HASH=5069692bfb
@@ -48,6 +48,17 @@ if [ ! -d "vlc" ]; then
     echo "Applying the patches"
     cd vlc
     git am ../patches/*.patch
+    git am ../../patches/vlc/*.patch
+elif [ '$1' != '-f' ]; then
+    echo "VLC source found, build without hash check"
+    cd vlc
+    echo "======================================="
+    echo "Applying the patches"
+    git am ../patches/*.patch || git am --abort
+    echo "======================================="
+    echo "Applying the patches for libvlc_danmaku"
+    git am ../../patches/vlc/*.patch || git am --abort
+    echo "======================================="
 else
     echo "VLC source found"
     cd vlc
@@ -64,6 +75,7 @@ git reset --hard origin
 git pull origin master
 git checkout -B android ${TESTED_HASH}
 git am ../patches/*
+git am ../../patches/vlc/*.patch
 
 *** : This will delete any changes you made to the current branch ***
 
@@ -72,6 +84,7 @@ EOF
     fi
 fi
 
+
 echo "Building the contribs"
 mkdir -p contrib/android
 cd contrib/android
@@ -102,6 +115,12 @@ if test -z "${NO_NEON}" -o -n "${TEGRA2}"; then
     echo "NOTHUMB := -marm" >> config.mak
 fi
 
+if  test -d "../../../../tarballs" ; then
+    echo "fetch local tarballs"
+    mkdir -p ../tarballs
+    cp -n -v ../../../../tarballs/* ../tarballs/ || true
+fi
+
 make fetch
 make
 
-- 
1.7.9.4

