From a9ace7f8304f20b3decf2c03d21e879d83ce2b1b Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 17:18:45 +0800
Subject: [PATCH] modify compile.sh for MacOS build

---
 compile.sh |   43 ++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 42 insertions(+), 1 deletion(-)

diff --git a/compile.sh b/compile.sh
index 8c2d614..a6c0c6b 100755
--- a/compile.sh
+++ b/compile.sh
@@ -37,7 +37,18 @@ fi
 
 # Add the NDK toolchain to the PATH, needed both for contribs and for building
 # stub libraries
-export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/linux-x86/bin:${PATH}
+BUILD_PLATFORM='unknown'
+if [[ "$(uname)" == 'Linux' ]]; then
+   BUILD_PLATFORM='linux'
+elif [[ "$(uname)" == 'Darwin' ]]; then
+   BUILD_PLATFORM='darwin'
+fi
+
+if [[ ${BUILD_PLATFORM} == 'darwin' ]]; then
+    export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/darwin-x86/bin:${PATH}
+elif [[ ${BUILD_PLATFORM} == 'linux' ]]; then
+    export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/linux-x86/bin:${PATH}
+fi
 
 # 1/ libvlc, libvlccore and its plugins
 TESTED_HASH=5069692bfb
@@ -48,6 +59,25 @@ if [ ! -d "vlc" ]; then
     echo "Applying the patches"
     cd vlc
     git am ../patches/*.patch
+
+    if [[ ${BUILD_PLATFORM} == 'darwin' ]]; then
+        git am ../../patches/vlc-on-macosx/*.patch
+    fi
+    git am ../../patches/vlc/*.patch
+
+elif [ '$1' != '-f' ]; then
+    echo "VLC source found, build without hash check"
+    cd vlc
+    echo "======================================="
+    echo "Applying the patches"
+    git am ../patches/*.patch || git am --abort
+    echo "======================================="
+    echo "Applying the patches for libvlc_danmaku"
+    if [[ ${BUILD_PLATFORM} == 'darwin' ]]; then
+        git am ../../patches/vlc-on-macosx/*.patch || git am --abort
+    fi
+    git am ../../patches/vlc/*.patch || git am --abort
+    echo "======================================="
 else
     echo "VLC source found"
     cd vlc
@@ -64,6 +94,10 @@ git reset --hard origin
 git pull origin master
 git checkout -B android ${TESTED_HASH}
 git am ../patches/*
+if [[ ${BUILD_PLATFORM} == 'darwin' ]]; then
+    git am ../../patches/vlc-on-macosx/*.patch
+fi
+git am ../../patches/vlc/*.patch
 
 *** : This will delete any changes you made to the current branch ***
 
@@ -72,6 +106,7 @@ EOF
     fi
 fi
 
+
 echo "Building the contribs"
 mkdir -p contrib/android
 cd contrib/android
@@ -102,6 +137,12 @@ if test -z "${NO_NEON}" -o -n "${TEGRA2}"; then
     echo "NOTHUMB := -marm" >> config.mak
 fi
 
+if  test -d "../../../../tarballs" ; then
+    echo "fetch local tarballs"
+    mkdir -p ../tarballs
+    cp -n -v ../../../../tarballs/* ../tarballs/ || true
+fi
+
 make fetch
 make
 
-- 
1.7.9.4

