From 992144816dd3542b12936dbf35567c9d192d1ab6 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 14 Jul 2012 15:15:42 +0800
Subject: [PATCH 01/13] .gitignore

---
 .gitignore | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index 32a12cc..6566438 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,7 @@
 /vlc/
 /android-libs/*.so
+/patches/100*.patch
+/pt
 local.properties
 .DS_Store
 *.keystore
-- 
1.8.3.4

