From dacd0e2152fde35b5306f8be68e1c70113b8aa18 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 17:18:45 +0800
Subject: [PATCH 1/2] modify compile.sh for MacOS build

---
 compile.sh |   11 ++++++++++-
 1 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/compile.sh b/compile.sh
index 28c5624..e9ed879 100755
--- a/compile.sh
+++ b/compile.sh
@@ -35,7 +35,7 @@ fi
 
 # Add the NDK toolchain to the PATH, needed both for contribs and for building
 # stub libraries
-export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/linux-x86/bin:${PATH}
+export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/darwin-x86/bin:${PATH}
 
 # 1/ libvlc, libvlccore and its plugins
 
@@ -53,6 +53,9 @@ echo "Applying the patches"
 cd vlc
 git am ../patches/*.patch || git am --abort
 
+echo "Applying the patches for MacOS"
+git am ../../patches/vlc/*.patch || git am --abort
+
 echo "Building the contribs"
 mkdir contrib/android; cd contrib/android
 ../bootstrap --host=arm-linux-androideabi --disable-disc --disable-sout --enable-small \
@@ -76,6 +79,12 @@ mkdir contrib/android; cd contrib/android
 
 # TODO: mpeg2, dts, theora
 
+if  test -d "../../../../tarballs" ; then
+    echo "fetch local tarballs"
+    mkdir -p ../tarballs
+    cp -n ../../../../tarballs/* ../tarballs/
+fi
+
 make fetch
 make
 
-- 
1.7.7.4

