From 2666ac314f3593606882a9fb187f2d0a4f71e15e Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 17:18:45 +0800
Subject: [PATCH 2/7] modify compile.sh for MacOS build

---
 compile.sh |    8 +++++++-
 1 个文件被修改，插入 7 行(+)，删除 1 行(-)

diff --git a/compile.sh b/compile.sh
index 355da1c..27dbb98 100755
--- a/compile.sh
+++ b/compile.sh
@@ -29,7 +29,7 @@ If you plan to use a release build, run 'compile.sh release'
 EOF
 # try to detect NDK version
 REL=$(grep -o '^r[0-9]*' $ANDROID_NDK/RELEASE.TXT 2>/dev/null|cut -b2-)
-if [ -z $REL ]; then
+if [ -z "$REL" ]; then
     echo "You need the NDKv7 or later"
     exit 1
 fi
@@ -162,6 +162,12 @@ fi
 echo "EXTRA_CFLAGS= -g ${EXTRA_CFLAGS}" >> config.mak
 export VLC_EXTRA_CFLAGS="${EXTRA_CFLAGS}"
 
+if  test -d "../../../../tarballs" ; then
+    echo "fetch local tarballs"
+    mkdir -p ../tarballs
+    cp -n -v ../../../../tarballs/* ../tarballs/ || true
+fi
+
 make fetch
 make
 
-- 
1.7.10.4

