From 5fafb888c09e1867c89b18c510a66410aa5bcae7 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 17:18:45 +0800
Subject: [PATCH 2/7] modify compile.sh for MacOS build

---
 compile.sh |    6 ++++++
 1 个文件被修改，插入 6 行(+)

diff --git a/compile.sh b/compile.sh
index fc01a26..64fa3ae 100755
--- a/compile.sh
+++ b/compile.sh
@@ -192,6 +192,12 @@ fi
 echo "EXTRA_CFLAGS= -g ${EXTRA_CFLAGS}" >> config.mak
 export VLC_EXTRA_CFLAGS="${EXTRA_CFLAGS}"
 
+if  test -d "../../../../tarballs" ; then
+    echo "fetch local tarballs"
+    mkdir -p ../tarballs
+    cp -n -v ../../../../tarballs/* ../tarballs/ || true
+fi
+
 make fetch
 make
 
-- 
1.7.10.4

