From 63b2eace06027dcdfb1d7e5f92520a141c51067e Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 11 Aug 2012 14:27:45 +0800
Subject: [PATCH 02/12] compile.sh: set CHOST for cross compile

---
 compile.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/compile.sh b/compile.sh
index 193966d..243c51e 100755
--- a/compile.sh
+++ b/compile.sh
@@ -79,6 +79,7 @@ export PLATFORM_SHORT_ARCH
 # Add the NDK toolchain to the PATH, needed both for contribs and for building
 # stub libraries
 NDK_TOOLCHAIN_PATH=`echo ${ANDROID_NDK}/toolchains/${PATH_HOST}-${GCCVER}/prebuilt/\`uname|tr A-Z a-z\`-*/bin`
+CHOST=${NDK_TOOLCHAIN_PATH}/${TARGET_TUPLE}
 export PATH=${NDK_TOOLCHAIN_PATH}:${PATH}
 
 ANDROID_PATH="`pwd`"
@@ -203,6 +204,7 @@ else
 fi
 
 echo "EXTRA_CFLAGS= -g ${EXTRA_CFLAGS}" >> config.mak
+echo "CHOST= ${CHOST}" >> config.mak
 export VLC_EXTRA_CFLAGS="${EXTRA_CFLAGS}"
 
 make fetch
-- 
1.8.3.4

