From ae41c1cc7221ba187e5e6fa8dcb31bc3383738f7 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 8 May 2012 11:30:23 +0800
Subject: [PATCH 2/5] do not build apk

---
 Makefile |    2 +-
 1 个文件被修改，插入 1 行(+)，删除 1 行(-)

diff --git a/Makefile b/Makefile
index 287e165..6d58a68 100644
--- a/Makefile
+++ b/Makefile
@@ -46,7 +46,7 @@ $(VLC_APK): $(LIBVLCJNI) $(JAVA_SOURCES)
 	date +"%Y-%m-%d" > $(SRC)/assets/builddate.txt
 	echo `id -u -n`@`hostname` > $(SRC)/assets/builder.txt
 	git rev-parse --short HEAD > $(SRC)/assets/revision.txt
-	$(VERBOSE)cd $(SRC) && ant $(ANT_OPTS) $(ANT_TARGET)
+	$(VERBOSE)cd $(SRC)
 
 VLC_MODULES=`./find_modules.sh $(VLC_BUILD_DIR)`
 
-- 
1.7.10.4

