From 7bfe9db01bc45eae981b97e4624d162595cde1b0 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 17:18:45 +0800
Subject: [PATCH 2/6] modify compile.sh for MacOS build

---
 compile.sh |   32 +++++++++++++++++++++++++++++++-
 1 个文件被修改，插入 31 行(+)，删除 1 行(-)

diff --git a/compile.sh b/compile.sh
index 34cff5f..efb1558 100755
--- a/compile.sh
+++ b/compile.sh
@@ -29,13 +29,24 @@ If you plan to use a release build, run 'compile.sh release'
 EOF
 # try to detect NDK version
 REL=$(grep -o '^r[0-9]*' $ANDROID_NDK/RELEASE.TXT 2>/dev/null|cut -b2-)
-if [ -z $REL ]; then
+if [ -z "$REL" ]; then
     echo "You need the NDKv7 or later"
     exit 1
 fi
 
 # Add the NDK toolchain to the PATH, needed both for contribs and for building
 # stub libraries
+BUILD_PLATFORM='unknown'
+if [ "$(uname)" == 'Linux' ]; then
+   BUILD_PLATFORM='linux'
+elif [ "$(uname)" == 'Darwin' ]; then
+   BUILD_PLATFORM='darwin'
+fi
+
+if [ ${BUILD_PLATFORM} == 'darwin' ]; then
+    export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/darwin-x86/bin:${PATH}
+fi
+
 export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/linux-x86/bin:${PATH}
 
 # 1/ libvlc, libvlccore and its plugins
@@ -47,6 +58,14 @@ if [ ! -d "vlc" ]; then
     git checkout -B android ${TESTED_HASH}
     echo "Applying the patches"
     git am ../patches/*.patch
+
+    if [ ${BUILD_PLATFORM} == 'darwin' ]; then
+        git am ../../patches/vlc-on-macosx/*.patch
+    fi
+    git am ../../patches/vlc/*.patch
+elif [ '$1' != '-f' ]; then
+    echo "VLC source found, build without hash check"
+    cd vlc
 else
     echo "VLC source found"
     cd vlc
@@ -63,6 +82,10 @@ git reset --hard origin
 git pull origin master
 git checkout -B android ${TESTED_HASH}
 git am ../patches/*
+if [[ ${BUILD_PLATFORM} == 'darwin' ]]; then
+    git am ../../patches/vlc-on-macosx/*.patch
+fi
+git am ../../patches/vlc/*.patch
 
 *** : This will delete any changes you made to the current branch ***
 
@@ -71,6 +94,7 @@ EOF
     fi
 fi
 
+
 echo "Building the contribs"
 mkdir -p contrib/android
 cd contrib/android
@@ -129,6 +153,12 @@ fi
 echo "EXTRA_CFLAGS= -g ${EXTRA_CFLAGS}" >> config.mak
 export VLC_EXTRA_CFLAGS="${EXTRA_CFLAGS}"
 
+if  test -d "../../../../tarballs" ; then
+    echo "fetch local tarballs"
+    mkdir -p ../tarballs
+    cp -n -v ../../../../tarballs/* ../tarballs/ || true
+fi
+
 make fetch
 make
 
-- 
1.7.10.4

