From 6ddc025b3bdfd6ef37c9aacf8dea3d790073e08f Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 17:18:45 +0800
Subject: [PATCH 3/8] compile.sh: restore saved contrib package

---
 compile.sh |    6 ++++++
 1 个文件被修改，插入 6 行(+)

diff --git a/compile.sh b/compile.sh
index 0f3af26..ceb3535 100755
--- a/compile.sh
+++ b/compile.sh
@@ -195,6 +195,12 @@ echo "EXTRA_CFLAGS= -g ${EXTRA_CFLAGS}" >> config.mak
 echp "CHOST= ${CHOST}" >> config.mak
 export VLC_EXTRA_CFLAGS="${EXTRA_CFLAGS}"
 
+if  test -d "../../../../tarballs" ; then
+    echo "fetch local tarballs"
+    mkdir -p ../tarballs
+    cp -n -v ../../../../tarballs/* ../tarballs/ || true
+fi
+
 make fetch
 make
 
-- 
1.7.10.4

