From d199088c17feb0dc18082261f7383d37dfb9fb31 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 8 May 2012 12:32:02 +0800
Subject: [PATCH 3/3] vlc repo reset script

---
 _reset_vlc.sh |   31 +++++++++++++++++++++++++++++++
 1 个文件被修改，插入 31 行(+)
 create mode 100644 _reset_vlc.sh

diff --git a/_reset_vlc.sh b/_reset_vlc.sh
new file mode 100644
index 0000000..69d9f41
--- /dev/null
+++ b/_reset_vlc.sh
@@ -0,0 +1,31 @@
+#! /bin/sh
+
+# 1/ libvlc, libvlccore and its plugins
+TESTED_HASH=6fdb0bea19
+if [ ! -d "vlc" ]; then
+    echo "VLC source not found, cloning"
+    git clone git://git.videolan.org/vlc.git vlc
+    cd vlc
+else
+    echo "VLC source found"
+    cd vlc
+    git reset --hard origin
+    git fetch origin
+fi
+
+echo "======================================="
+echo "Reset to brance android ${TESTED_HASH}"
+git checkout -B android ${TESTED_HASH}
+
+echo "======================================="
+echo "Applying the patches"
+git am ../patches/*.patch || git am --abort
+
+echo "======================================="
+echo "Applying the patches for libvlc_danmaku"
+git am ../../patches/vlc/*.patch || git am --abort
+git am ../patches-extra/*.patch || git am --abort
+
+echo "======================================="
+
+cd -
-- 
1.7.10.2

