From e645a5df99941efea61357750468a0c0ffd7d572 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 17:18:45 +0800
Subject: [PATCH 03/13] compile.sh: restore saved contrib package

---
 compile.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/compile.sh b/compile.sh
index 68335c1..660968e 100755
--- a/compile.sh
+++ b/compile.sh
@@ -255,6 +255,12 @@ echo "EXTRA_CFLAGS= -g ${EXTRA_CFLAGS}" >> config.mak
 echo "CHOST= ${CHOST}" >> config.mak
 export VLC_EXTRA_CFLAGS="${EXTRA_CFLAGS}"
 
+if  test -d "../../../../tarballs" ; then
+    echo "fetch local tarballs"
+    mkdir -p ../tarballs
+    cp -n -v ../../../../tarballs/* ../tarballs/ || true
+fi
+
 make fetch
 # We already have zlib available
 [ -e .zlib ] || (mkdir -p zlib; touch .zlib)
-- 
1.8.3.4

