From d5c5e98c1da375add1b55b56a0eb6e5c654afdb8 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 8 May 2012 11:30:23 +0800
Subject: [PATCH 3/6] do not build apk

---
 Makefile |    2 +-
 1 个文件被修改，插入 1 行(+)，删除 1 行(-)

diff --git a/Makefile b/Makefile
index ca687a8..48e0540 100644
--- a/Makefile
+++ b/Makefile
@@ -47,7 +47,7 @@ $(VLC_APK): $(LIBVLCJNI) $(JAVA_SOURCES)
 	echo `id -u -n`@`hostname` > $(SRC)/assets/builder.txt
 	git rev-parse --short HEAD > $(SRC)/assets/revision.txt
 	./gen-env.sh $(SRC)
-	$(VERBOSE)cd $(SRC) && ant $(ANT_OPTS) $(ANT_TARGET)
+	$(VERBOSE)cd $(SRC)
 
 VLC_MODULES=`./find_modules.sh $(VLC_BUILD_DIR)`
 
-- 
1.7.10.4

