From 1f32734c716e6a90d35152b2d7e103335027f41c Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 5 Mar 2012 17:18:45 +0800
Subject: [PATCH 3/8] compile.sh: restore saved contrib package

---
 compile.sh |    6 ++++++
 1 个文件被修改，插入 6 行(+)

diff --git a/compile.sh b/compile.sh
index c144c30..2721901 100755
--- a/compile.sh
+++ b/compile.sh
@@ -191,6 +191,12 @@ echo "EXTRA_CFLAGS= -g ${EXTRA_CFLAGS}" >> config.mak
 echo "CHOST= ${CHOST}" >> config.mak
 export VLC_EXTRA_CFLAGS="${EXTRA_CFLAGS}"
 
+if  test -d "../../../../tarballs" ; then
+    echo "fetch local tarballs"
+    mkdir -p ../tarballs
+    cp -n -v ../../../../tarballs/* ../tarballs/ || true
+fi
+
 make fetch
 make $MAKEFLAGS
 
-- 
1.7.10.4

