From a73c40b8654bfa8481a6d280ad00d595ada788be Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 28 Jul 2012 17:21:40 +0800
Subject: [PATCH 4/8] compile.sh: support fastcompile.sh

---
 compile.sh     |   12 ++++++++++++
 fastcompile.sh |    4 ++++
 2 个文件被修改，插入 16 行(+)
 create mode 100644 fastcompile.sh

diff --git a/compile.sh b/compile.sh
index ceb3535..1f2ea1f 100755
--- a/compile.sh
+++ b/compile.sh
@@ -90,6 +90,7 @@ CHOST=${ANDROID_BIN}/${TARGET_TUPLE}
 export PATH=${ANDROID_BIN}:${PATH}
 
 # 1/ libvlc, libvlccore and its plugins
+if [ -z "${VLC_FAST_COMPILE}" ]; then
 TESTED_HASH=9eb524d
 if [ ! -d "vlc" ]; then
     echo "VLC source not found, cloning"
@@ -126,6 +127,9 @@ EOF
         exit 1
     fi
 fi
+else
+    cd vlc
+fi
 
 if [ ${ANDROID_ABI} = "armeabi-v7a" ] ; then
     if test -z "${NO_NEON}" ; then
@@ -158,6 +162,7 @@ EXTRA_CFLAGS="${EXTRA_CFLAGS} -I${ANDROID_NDK}/sources/cxx-stl/gnu-libstdc++${CX
 echo "Building the contribs"
 mkdir -p contrib/android
 cd contrib/android
+if [ -z "${VLC_FAST_COMPILE}" ]; then
 ../bootstrap --host=${TARGET_TUPLE} --disable-disc --disable-sout --enable-small \
     --disable-sdl \
     --disable-SDL_image \
@@ -175,6 +180,7 @@ cd contrib/android
     --disable-samplerate \
     --disable-faad2 \
     --enable-iconv
+fi
 
 # TODO: mpeg2, theora
 
@@ -191,10 +197,13 @@ else
     OPTS="--enable-debug"
 fi
 
+if [ -z "${VLC_FAST_COMPILE}" ]; then
 echo "EXTRA_CFLAGS= -g ${EXTRA_CFLAGS}" >> config.mak
 echp "CHOST= ${CHOST}" >> config.mak
+fi
 export VLC_EXTRA_CFLAGS="${EXTRA_CFLAGS}"
 
+if [ -z "${VLC_FAST_COMPILE}" ]; then
 if  test -d "../../../../tarballs" ; then
     echo "fetch local tarballs"
     mkdir -p ../tarballs
@@ -203,9 +212,11 @@ fi
 
 make fetch
 make
+fi
 
 cd ../.. && mkdir -p android && cd android
 
+if [ -z "${VLC_FAST_COMPILE}" ]; then
 if test ! -s "../configure" ; then
     echo "Bootstraping"
     ../bootstrap
@@ -213,6 +224,7 @@ fi
 
 echo "Configuring"
 ../../configure.sh $OPTS
+fi
 
 echo "Building"
 make
diff --git a/fastcompile.sh b/fastcompile.sh
new file mode 100644
index 0000000..97130c1
--- /dev/null
+++ b/fastcompile.sh
@@ -0,0 +1,4 @@
+#! /bin/sh
+
+export VLC_FAST_COMPILE=1
+sh compile.sh
-- 
1.7.10.4

