From b4f6ac5a111471bf8ae0cb28432a4379f507bbf4 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 16 Apr 2012 23:20:57 +0800
Subject: [PATCH 4/8] disable modules in configure.sh

---
 configure.sh |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.sh b/configure.sh
index fd313f9..ab51923 100755
--- a/configure.sh
+++ b/configure.sh
@@ -46,7 +46,7 @@ STRIP="${CROSS_COMPILE}strip" \
 RANLIB="${CROSS_COMPILE}ranlib" \
 AR="${CROSS_COMPILE}ar" \
 sh $VLC_SOURCEDIR/configure --host=arm-linux-androideabi --build=x86_64-unknown-linux $EXTRA_PARAMS \
-                --enable-live555 --enable-realrtsp \
+                --disable-live555 --disable-realrtsp \
                 --enable-avformat \
                 --enable-swscale \
                 --enable-avcodec \
@@ -54,8 +54,8 @@ sh $VLC_SOURCEDIR/configure --host=arm-linux-androideabi --build=x86_64-unknown-
                 --enable-audiotrack \
                 --enable-android-surface \
                 --enable-debug \
-                --enable-mkv \
-                --enable-taglib \
+                --disable-mkv \
+                --disable-taglib \
                 --enable-iomx \
                 --disable-vlc --disable-shared \
                 --disable-vlm \
-- 
1.7.9.4

