From cb92c15c451abf4ee155b80bf5c04df0380efde0 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 28 Jul 2012 17:21:40 +0800
Subject: [PATCH] compile.sh: support fastcompile.sh

---
 fastcompile.sh | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 fastcompile.sh

diff --git a/fastcompile.sh b/fastcompile.sh
new file mode 100644
index 0000000..8617530
--- /dev/null
+++ b/fastcompile.sh
@@ -0,0 +1,4 @@
+#! /bin/sh
+
+export VLC_FAST_COMPILE=1
+sh compile.sh --build
-- 
1.8.3.4

