From 8ccc4df69a3d822cbaaf76f2bc7ed31f382ca6fa Mon Sep 17 00:00:00 2001
From: Ming Hu <tewilove@gmail.com>
Date: Thu, 23 Oct 2014 15:44:54 +0800
Subject: [PATCH 05/12] libvlcjni-util.c: fix leak of Java class local ref

---
 vlc-android/jni/libvlcjni-util.c | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/vlc-android/jni/libvlcjni-util.c b/vlc-android/jni/libvlcjni-util.c
index 205d59d..b4a4492 100644
--- a/vlc-android/jni/libvlcjni-util.c
+++ b/vlc-android/jni/libvlcjni-util.c
@@ -45,7 +45,9 @@ jint getInt(JNIEnv *env, jobject thiz, const char* field) {
     jclass clazz = (*env)->GetObjectClass(env, thiz);
     jfieldID fieldMP = (*env)->GetFieldID(env, clazz,
                                           field, "I");
-    return (*env)->GetIntField(env, thiz, fieldMP);
+    jint ret = (*env)->GetIntField(env, thiz, fieldMP);
+    (*env)->DeleteLocalRef(env, clazz);
+    return ret;
 }
 void setInt(JNIEnv *env, jobject item, const char* field, jint value) {
     jclass cls;
@@ -60,13 +62,16 @@ void setInt(JNIEnv *env, jobject item, const char* field, jint value) {
         return;
 
     (*env)->SetIntField(env, item, fieldId, value);
+    (*env)->DeleteLocalRef(env, cls);
 }
 
 jlong getLong(JNIEnv *env, jobject thiz, const char* field) {
     jclass clazz = (*env)->GetObjectClass(env, thiz);
     jfieldID fieldMP = (*env)->GetFieldID(env, clazz,
                                           field, "J");
-    return (*env)->GetLongField(env, thiz, fieldMP);
+    jlong ret = (*env)->GetLongField(env, thiz, fieldMP);
+    (*env)->DeleteLocalRef(env, clazz);
+    return ret;
 }
 void setLong(JNIEnv *env, jobject item, const char* field, jlong value) {
     jclass cls;
@@ -81,6 +86,7 @@ void setLong(JNIEnv *env, jobject item, const char* field, jlong value) {
         return;
 
     (*env)->SetLongField(env, item, fieldId, value);
+    (*env)->DeleteLocalRef(env, cls);
 }
 
 void setFloat(JNIEnv *env, jobject item, const char* field, jfloat value) {
@@ -96,6 +102,7 @@ void setFloat(JNIEnv *env, jobject item, const char* field, jfloat value) {
         return;
 
     (*env)->SetFloatField(env, item, fieldId, value);
+    (*env)->DeleteLocalRef(env, cls);
 }
 void setString(JNIEnv *env, jobject item, const char* field, const char* text) {
     jclass cls;
@@ -115,6 +122,7 @@ void setString(JNIEnv *env, jobject item, const char* field, const char* text) {
     if (jstr == NULL)
         return;
     (*env)->SetObjectField(env, item, fieldId, jstr);
+    (*env)->DeleteLocalRef(env, cls);
 }
 
 void arrayListGetIDs(JNIEnv *env, jclass* p_class, jmethodID* p_add, jmethodID* p_remove) {
@@ -142,10 +150,13 @@ jobject getEventHandlerReference(JNIEnv *env, jobject thiz, jobject eventHandler
     jmethodID methodID = (*env)->GetMethodID(env, cls, "callback", "(ILandroid/os/Bundle;)V");
     if (!methodID) {
         LOGE("setEventHandler: failed to get the callback method");
+        (*env)->DeleteLocalRef(env, cls);
         return NULL;
     }
 
-    return (*env)->NewGlobalRef(env, eventHandler);
+    jobject ref = (*env)->NewGlobalRef(env, eventHandler);
+    (*env)->DeleteLocalRef(env, cls);
+    return ref;
 }
 
 static void debug_buffer_log(void *data, int level, const char *fmt, va_list ap)
-- 
2.1.3

