From 932f874238497e3f294bd0501a0630487810fac5 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 8 May 2012 11:30:23 +0800
Subject: [PATCH 5/8] do not build apk

---
 Makefile |    2 +-
 1 个文件被修改，插入 1 行(+)，删除 1 行(-)

diff --git a/Makefile b/Makefile
index 7f6e946..9d512a6 100644
--- a/Makefile
+++ b/Makefile
@@ -43,7 +43,7 @@ $(VLC_APK): $(LIBVLCJNI) $(JAVA_SOURCES)
 	echo `id -u -n`@`hostname` > $(SRC)/assets/builder.txt
 	git rev-parse --short HEAD > $(SRC)/assets/revision.txt
 	./gen-env.sh $(SRC)
-	$(VERBOSE)cd $(SRC) && ant $(ANT_OPTS) $(ANT_TARGET)
+	$(VERBOSE)cd $(SRC)
 
 VLC_MODULES=`./find_modules.sh $(VLC_BUILD_DIR)`
 
-- 
1.7.10.4

