From 1bc7d48a940c8c209fa7c7b3b579191e2c893637 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 20 May 2013 22:29:57 +0800
Subject: [PATCH 05/13] compile.sh: disable extras/tools

---
 compile.sh | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/compile.sh b/compile.sh
index 660968e..41d4734 100755
--- a/compile.sh
+++ b/compile.sh
@@ -187,12 +187,12 @@ then
 MAKEFLAGS=-j`sysctl -n machdep.cpu.thread_count`
 fi
 
-export PATH=`pwd`/extras/tools/build/bin:$PATH
-echo "Building tools"
-cd extras/tools
-./bootstrap
-make $MAKEFLAGS
-cd ../..
+# export PATH=`pwd`/extras/tools/build/bin:$PATH
+# echo "Building tools"
+# cd extras/tools
+# ./bootstrap
+# make $MAKEFLAGS
+# cd ../..
 
 echo "Building the contribs"
 mkdir -p contrib/android
-- 
1.8.3.4

