From 6aa41467ea402af4ce05ea49381c140aa8b882fa Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 20 May 2013 22:29:57 +0800
Subject: [PATCH 05/12] compile.sh: disable extras/tools

---
 compile.sh | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/compile.sh b/compile.sh
index 7570011..3e8c348 100755
--- a/compile.sh
+++ b/compile.sh
@@ -158,12 +158,12 @@ then
 MAKEFLAGS=-j`sysctl -n machdep.cpu.thread_count`
 fi
 
-export PATH=`pwd`/extras/tools/build/bin:$PATH
-echo "Building tools"
-cd extras/tools
-./bootstrap
-make $MAKEFLAGS
-cd ../..
+# export PATH=`pwd`/extras/tools/build/bin:$PATH
+# echo "Building tools"
+# cd extras/tools
+# ./bootstrap
+# make $MAKEFLAGS
+# cd ../..
 
 echo "Building the contribs"
 mkdir -p contrib/android
-- 
1.8.3.4

