From 2b2808199abcdaa5fc48f8a3b344f5b573ad1c49 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 24 Mar 2012 16:55:12 +0800
Subject: [PATCH 6/6] disable schroedinger

---
 compile.sh |    1 +
 1 file changed, 1 insertion(+)

diff --git a/compile.sh b/compile.sh
index 7ba2e88..cbed62b 100755
--- a/compile.sh
+++ b/compile.sh
@@ -79,6 +79,7 @@ mkdir contrib/android; cd contrib/android
     --disable-sidplay2 \
     --disable-taglib \
     --disable-live555 \
+    --disable-schroedinger \
     --disable-samplerate
 
 # TODO: mpeg2, dts, theora
-- 
1.7.9.4

