From 9e1c7ed86b7048821ea40edb3bd52e01a11c4761 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 8 May 2012 12:32:02 +0800
Subject: [PATCH 6/8] vlc repo reset script

---
 _reset_vlc.sh |    3 +++
 1 个文件被修改，插入 3 行(+)
 create mode 100644 _reset_vlc.sh

diff --git a/_reset_vlc.sh b/_reset_vlc.sh
new file mode 100644
index 0000000..f576ca8
--- /dev/null
+++ b/_reset_vlc.sh
@@ -0,0 +1,3 @@
+#! /bin/sh
+
+sh ../_do_reset_vlc.sh
-- 
1.7.10.4

