From 9f68306aec35f7cea516740dd825725051a8ac70 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 15 May 2012 17:37:16 +0800
Subject: [PATCH 6/6] update vlc to 0ef62578e5b6

---
 _reset_vlc.sh |    2 +-
 1 个文件被修改，插入 1 行(+)，删除 1 行(-)

diff --git a/_reset_vlc.sh b/_reset_vlc.sh
index 4c56ed9..b25f7bf 100644
--- a/_reset_vlc.sh
+++ b/_reset_vlc.sh
@@ -1,7 +1,7 @@
 #! /bin/sh
 
 # 1/ libvlc, libvlccore and its plugins
-TESTED_HASH=a685fdeb8d
+TESTED_HASH=0ef62578e5b6
 if [ ! -d "vlc" ]; then
     echo "VLC source not found, cloning"
     git clone git://git.videolan.org/vlc.git vlc
-- 
1.7.10.2

