From ccd0a93c0b57a2bb42a6ab8377ce9ff53b1ceed6 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 8 May 2012 11:30:23 +0800
Subject: [PATCH 07/16] Makefile: do not build apk

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8ab1ace..34d2af6 100644
--- a/Makefile
+++ b/Makefile
@@ -45,7 +45,7 @@ $(VLC_APK): $(LIBVLCJNI) $(JAVA_SOURCES)
 	echo `id -u -n`@`hostname` > $(SRC)/assets/builder.txt
 	git rev-parse --short HEAD > $(SRC)/assets/revision.txt
 	./gen-env.sh $(SRC)
-	$(VERBOSE)cd $(SRC) && ant $(ANT_OPTS) $(ANT_TARGET)
+	$(VERBOSE)cd $(SRC)
 
 VLC_MODULES=`./find_modules.sh $(VLC_BUILD_DIR)`
 
-- 
1.9.0

