From c24c1f21141f2038d7974afff0dd79cd62e7c6f3 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 26 Apr 2012 15:27:30 +0800
Subject: [PATCH 7/7] update vlc hash

---
 compile.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compile.sh b/compile.sh
index 0165277..fd4c428 100755
--- a/compile.sh
+++ b/compile.sh
@@ -38,7 +38,7 @@ fi
 export PATH=${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.4.3/prebuilt/darwin-x86/bin:${PATH}
 
 # 1/ libvlc, libvlccore and its plugins
-TESTED_HASH=5069692bfb
+TESTED_HASH=a685fdeb8d
 if [ ! -d "vlc" ]; then
     echo "VLC source not found, cloning"
     git clone git://git.videolan.org/vlc.git vlc
-- 
1.7.9.4

