From 1f5c90336a1ac2c24e8229e7e6bcfe275cd7cebe Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Tue, 28 Jan 2014 13:52:42 +0800
Subject: [PATCH 08/12] Android.mk: include contrib

---
 vlc-android/jni/Android.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/vlc-android/jni/Android.mk b/vlc-android/jni/Android.mk
index 87dea03..2ba07ce 100644
--- a/vlc-android/jni/Android.mk
+++ b/vlc-android/jni/Android.mk
@@ -39,6 +39,7 @@ LOCAL_SRC_FILES += wchar/wmemset.c
 
 
 LOCAL_C_INCLUDES := $(VLC_SRC_DIR)/include
+LOCAL_C_INCLUDES += $(VLC_SRC_DIR)/contrib/arm-linux-androideabi/include
 
 ARCH=$(ANDROID_ABI)
 
-- 
1.8.3.4

