From 0d519c3c2a6cf095da8909eb52c896d5071879de Mon Sep 17 00:00:00 2001
From: Ming Hu <tewilove@gmail.com>
Date: Fri, 31 Oct 2014 11:18:50 +0800
Subject: [PATCH 08/12] find_modules: correct a name

---
 find_modules.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/find_modules.sh b/find_modules.sh
index 6e70bb8..3885799 100755
--- a/find_modules.sh
+++ b/find_modules.sh
@@ -45,7 +45,7 @@ smf
 podcast
 bluescreen
 erase
-stream_filter_record
+record
 speex_resampler
 remoteosd
 magnify
-- 
2.1.3

