From 4376808cafb2135b7a51e474f47d9e328bdacb62 Mon Sep 17 00:00:00 2001
From: Ming Hu <tewilove@gmail.com>
Date: Fri, 31 Oct 2014 11:38:10 +0800
Subject: [PATCH 09/12] Android.mk: link less libs

---
 vlc-android/jni/Android.mk | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/vlc-android/jni/Android.mk b/vlc-android/jni/Android.mk
index a73bf97..8a903b1 100644
--- a/vlc-android/jni/Android.mk
+++ b/vlc-android/jni/Android.mk
@@ -67,19 +67,8 @@ LOCAL_LDLIBS := -L$(VLC_CONTRIB)/lib \
 	$(VLC_BUILD_DIR)/src/.libs/libvlccore.a \
 	$(VLC_BUILD_DIR)/compat/.libs/libcompat.a \
 	-ldl -lz -lm -llog \
-	-ldvbpsi -lebml -lmatroska -ltag \
-	-logg -lFLAC -ltheora -lvorbis \
-	-lmpeg2 -la52 \
-	-lavformat -lavcodec -lswscale -lavutil -lpostproc -lgsm -lopenjpeg \
-	-lliveMedia -lUsageEnvironment -lBasicUsageEnvironment -lgroupsock \
-	-lspeex -lspeexdsp \
-	-lxml2 -lpng -lgnutls -lgcrypt -lgpg-error \
-	-lnettle -lhogweed -lgmp \
-	-lfreetype -liconv -lass -lfribidi -lopus \
-	-lEGL -lGLESv2 -ljpeg \
-	-ldvdnav -ldvdread -ldvdcss \
-	-lmad \
-	$(CPP_STATIC)
+	-lijkffmpeg \
+	-lEGL -lGLESv2
 
 include $(BUILD_SHARED_LIBRARY)
 
-- 
2.1.3

