From b980e49d88e43be71297f346710935f6be5944c2 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 15 Oct 2012 14:14:55 +0800
Subject: [PATCH 10/10] Application.mk: use gcc 4.4.3

---
 vlc-android/jni/Application.mk |    2 +-
 1 个文件被修改，插入 1 行(+)，删除 1 行(-)

diff --git a/vlc-android/jni/Application.mk b/vlc-android/jni/Application.mk
index 0bd8e8a..006c98a 100644
--- a/vlc-android/jni/Application.mk
+++ b/vlc-android/jni/Application.mk
@@ -1,3 +1,3 @@
 APP_PLATFORM := android-9
 APP_ABI := $(ANDROID_ABI)
-NDK_TOOLCHAIN_VERSION=4.6
+NDK_TOOLCHAIN_VERSION=4.4.3
-- 
1.7.10.4

