From 549a1c580c220bed9c1162941e104469dc4247bb Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 19 Oct 2013 14:32:32 +0800
Subject: [PATCH 13/14] aout: limit sample rate to 48kHz

---
 vlc-android/jni/aout.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/vlc-android/jni/aout.c b/vlc-android/jni/aout.c
index 859509b..a8ae95d 100644
--- a/vlc-android/jni/aout.c
+++ b/vlc-android/jni/aout.c
@@ -79,6 +79,9 @@ int aout_open(void **opaque, char *format, unsigned *rate, unsigned *nb_channels
     LOGV ("Number of channels forced to 2, number of samples to %d", FRAME_SIZE);
     *nb_channels = 2;
 
+    if (*rate > 48000)
+        *rate = 48000;
+
     (*p_env)->CallVoidMethod (p_env, p_sys->j_libVlc, methodIdInitAout,
                               *rate, *nb_channels, FRAME_SIZE);
     if ((*p_env)->ExceptionCheck (p_env))
-- 
1.8.3.4

