From 730ef15d9cbaf4929e345661030bee1c91ca4319 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sat, 24 Aug 2013 00:18:13 +0800
Subject: [PATCH] compile.sh: support NDKr9

---
 compile.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/compile.sh b/compile.sh
index cbea322..5383fda 100755
--- a/compile.sh
+++ b/compile.sh
@@ -18,6 +18,10 @@ fi
 # try to detect NDK version
 REL=$(grep -o '^r[0-9]*.*' $ANDROID_NDK/RELEASE.TXT 2>/dev/null|cut -b2-)
 case "$REL" in
+    9*)
+        GCCVER=4.8
+        CXXSTL="/"${GCCVER}
+    ;;
     8?*)
         # we don't use 4.4.3 because it doesn't handle threads correctly.
         # TODO : clang?
-- 
1.8.3.4

