From e131a76ecbfbc6f27690279f4d0d46e37f639189 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Fri, 14 Feb 2014 17:18:01 +0800
Subject: [PATCH] compile.sh: fix build jni for target x86

---
 compile.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compile.sh b/compile.sh
index 73675a4..a07552e 100755
--- a/compile.sh
+++ b/compile.sh
@@ -266,7 +266,7 @@ cd ../.. && mkdir -p android && cd android
 
 if [ $# -eq 1 ] && [ "$1" = "jni" ]; then
     CLEAN="jniclean"
-    TARGET="vlc-android/obj/local/armeabi-v7a/libvlcjni.so"
+    TARGET="vlc-android/obj/local/$ANDROID_ABI/libvlcjni.so"
 else
     CLEAN="distclean"
     if [ ! -f ../configure ]; then
-- 
1.8.5.2 (Apple Git-48)

