From 1cafa414626346b14b6ea09177204cb29f5d6bcb Mon Sep 17 00:00:00 2001
From: Ming Hu <tewilove@gmail.com>
Date: Tue, 10 Jun 2014 10:08:18 +0800
Subject: [PATCH] correct a name

---
 find_modules.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/find_modules.sh b/find_modules.sh
index 0be2ff5..67f6085 100755
--- a/find_modules.sh
+++ b/find_modules.sh
@@ -45,7 +45,7 @@ smf
 podcast
 bluescreen
 erase
-stream_filter_record
+record
 speex_resampler
 remoteosd
 magnify
-- 
1.8.5.2 (Apple Git-48)

