From c41c6eff639c19bb0900c215811b09a19c93a853 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 24 Jun 2012 17:23:34 +0800
Subject: [PATCH] build for armeabi-v7a with NEON

---
 compile.sh |    3 +++
 1 个文件被修改，插入 3 行(+)

diff --git a/compile.sh b/compile.sh
index 3b44def..d02a167 100755
--- a/compile.sh
+++ b/compile.sh
@@ -6,6 +6,9 @@
 
 set -e
 
+# patched for armeabi-v7a with NEON
+export ANDROID_ABI=armeabi-v7a
+
 if [ -z "$ANDROID_NDK" -o -z "$ANDROID_SDK" -o -z "$ANDROID_ABI" ]; then
    echo "You must define ANDROID_NDK, ANDROID_SDK and ANDROID_ABI before starting."
    echo "They must point to your NDK and SDK directories.\n"
-- 
1.7.10.4

