From f9464db6793d3774c096d73f7075113f253aaad3 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Fri, 23 Mar 2012 17:59:42 +0800
Subject: [PATCH 1/4] medium-mode: Android.mk: disable modules

---
 vlc-android/jni/Android.mk |   11 ++++-------
 1 个文件被修改，插入 4 行(+)，删除 7 行(-)

diff --git a/vlc-android/jni/Android.mk b/vlc-android/jni/Android.mk
index f9efd9e..d94275a 100644
--- a/vlc-android/jni/Android.mk
+++ b/vlc-android/jni/Android.mk
@@ -31,14 +31,11 @@ LOCAL_LDLIBS := -L$(VLC_CONTRIB)/lib \
 	$(VLC_BUILD_DIR)/src/.libs/libvlccore.a \
 	$(VLC_BUILD_DIR)/compat/.libs/libcompat.a \
 	-ldl -lz -lm -llog \
-	-ldvbpsi -lebml -lmatroska -ltag \
-	-logg -lFLAC -ltheora \
-	-lmpeg2 -ldca -la52 \
-	-lavformat -lavcodec -lswscale -lavutil -lpostproc -lgsm -lopenjpeg \
 	-lliveMedia -lUsageEnvironment -lBasicUsageEnvironment -lgroupsock \
-	-lspeex -lspeexdsp \
-	-lxml2 -lpng -lgnutls -lgcrypt -lgpg-error \
-	-lfreetype -liconv -lass -lfribidi \
+	-ldvbpsi \
+	-lmpeg2 -la52 \
+	-lavformat -lavcodec -lswscale -lavutil -lpostproc \
+	-lxml2 -lgnutls -lgcrypt -lgpg-error \
 	$(CPP_STATIC)
 
 include $(BUILD_SHARED_LIBRARY)
-- 
1.7.10.4

