From b299af9d2c19810c469f4b47428842e95e6db7a3 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 28 May 2012 11:45:12 +0800
Subject: [PATCH 2/4] medium-mode: compile.sh: disable contrib

---
 compile.sh |   19 +++++++++++++++++++
 1 个文件被修改，插入 19 行(+)

diff --git a/compile.sh b/compile.sh
index 4be7ab0..3d320aa 100755
--- a/compile.sh
+++ b/compile.sh
@@ -115,6 +115,8 @@ if [ -z "${VLC_FAST_COMPILE}" ]; then
     --disable-sdl \
     --disable-SDL_image \
     --disable-fontconfig \
+    --disable-ass \
+    --disable-fribidi \
     --disable-zvbi \
     --disable-kate \
     --disable-caca \
@@ -125,6 +127,23 @@ if [ -z "${VLC_FAST_COMPILE}" ]; then
     --disable-tremor \
     --disable-vorbis \
     --disable-sidplay2 \
+    --disable-taglib \
+    --enable-live555 \
+    --disable-schroedinger \
+    --disable-matroska \
+    --disable-goom \
+    --disable-gsm \
+    --disable-speex \
+    --disable-speexdsp \
+    --disable-freetype2 \
+    --disable-lua \
+    --disable-flac \
+    --disable-ogg \
+    --disable-dca \
+    --disable-theora \
+    --disable-modplug \
+    --disable-png \
+    --disable-openjpg \
     --disable-samplerate \
     --disable-faad2 \
     --disable-iconv
-- 
1.7.10.4

