From 8ab67c4fa30b2423a5877adb7b4c145453b5e88f Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Mon, 28 May 2012 11:45:12 +0800
Subject: [PATCH 2/4] medium-mode: compile.sh: disable contrib

---
 compile.sh |   21 ++++++++++++++++++++-
 1 个文件被修改，插入 20 行(+)，删除 1 行(-)

diff --git a/compile.sh b/compile.sh
index 718a1bc..af225fa 100755
--- a/compile.sh
+++ b/compile.sh
@@ -105,6 +105,8 @@ cd contrib/android
     --disable-sdl \
     --disable-SDL_image \
     --disable-fontconfig \
+    --disable-ass \
+    --disable-fribidi \
     --disable-zvbi \
     --disable-kate \
     --disable-caca \
@@ -115,9 +117,26 @@ cd contrib/android
     --disable-tremor \
     --disable-vorbis \
     --disable-sidplay2 \
+    --disable-taglib \
+    --enable-live555 \
+    --disable-schroedinger \
+    --disable-matroska \
+    --disable-goom \
+    --disable-gsm \
+    --disable-speex \
+    --disable-speexdsp \
+    --disable-freetype2 \
+    --disable-lua \
+    --disable-flac \
+    --disable-ogg \
+    --disable-dca \
+    --disable-theora \
+    --disable-modplug \
+    --disable-png \
+    --disable-openjpg \
     --disable-samplerate \
     --disable-faad2 \
-    --enable-iconv
+    --disable-iconv
 
 # TODO: mpeg2, theora
 
-- 
1.7.10.4

