From 0df53e2dc975aa714bb9436777eccb5d28ecfac3 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Thu, 22 Mar 2012 18:45:25 +0800
Subject: [PATCH 1001/1002] medium-mode: ffmpeg: disable modules

---
 contrib/src/ffmpeg/rules.mak |   48 +++++++++++++++++++++++++++++++++++++++---
 1 个文件被修改，插入 45 行(+)，删除 3 行(-)

diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index ed6d8e8..c8fda76 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -6,10 +6,52 @@ FFMPEG_SNAPURL := http://git.libav.org/?p=libav.git;a=snapshot;h=HEAD;sf=tgz
 FFMPEGCONF = \
 	--cc="$(CC)" \
 	--disable-doc \
+	--disable-network \
+	--disable-protocols \
 	--disable-decoder=libvpx \
 	--disable-decoder=bink \
-	--enable-libgsm \
-	--enable-libopenjpeg \
+	--disable-decoders \
+	--enable-decoder=aac \
+	--enable-decoder=aac_latm \
+	--enable-decoder=aasc \
+	--enable-decoder=ac3 \
+	--enable-decoder=amrnb \
+	--enable-decoder=amrwb \
+	--enable-decoder=flv \
+	--enable-decoder=h263 \
+	--enable-decoder=h263i \
+	--enable-decoder=h264 \
+	--enable-decoder=mp3 \
+	--enable-decoder=mp3adu \
+	--enable-decoder=mp3adufloat \
+	--enable-decoder=mp3float \
+	--enable-decoder=mp3on4 \
+	--enable-decoder=mp3on4float \
+	--enable-decoder=mpeg1video \
+	--enable-decoder=mpeg2video \
+	--enable-decoder=mpeg4 \
+	--enable-decoder=mpeg4v1 \
+	--enable-decoder=mpeg4v2 \
+	--enable-decoder=mpeg4v3 \
+	--enable-decoder=wmalossless \
+	--enable-decoder=wmapro \
+	--enable-decoder=wmav1 \
+	--enable-decoder=wmav2 \
+	--enable-decoder=wmavoice \
+	--enable-decoder=wmv1 \
+	--enable-decoder=wmv2 \
+	--enable-decoder=wmv3 \
+	--enable-decoder=wmv3_vdpau \
+	--enable-decoder=wmv3image \
+	--enable-decoder=vp3 \
+	--enable-decoder=vp5 \
+	--enable-decoder=vp6 \
+	--enable-decoder=vp6a \
+	--enable-decoder=vp6f \
+	--enable-decoder=vp8 \
+	--enable-decoder=vc1 \
+	--enable-decoder=vc1_vdpau \
+	--enable-decoder=vc1image \
 	--disable-debug \
 	--disable-avdevice \
 	--disable-devices \
@@ -20,7 +62,7 @@ FFMPEGCONF = \
 #	--disable-ffserver \
 #	--disable-ffplay \
 #	--disable-ffprobe
-DEPS_ffmpeg = zlib gsm openjpeg
+DEPS_ffmpeg = zlib
 
 # Optional dependencies
 ifdef BUILD_ENCODERS
-- 
1.7.10.4

