From e34ca373b39c32130fb1a4ad4721bb3b14cd25ef Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 24 Jun 2012 17:23:34 +0800
Subject: [PATCH] build for armeabi-v7a

---
 compile.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/compile.sh b/compile.sh
index d025c2e..c412e46 100755
--- a/compile.sh
+++ b/compile.sh
@@ -6,6 +6,9 @@
 
 set -e
 
+# patched for armeabi-v7a
+export ANDROID_ABI=armeabi-v7a
+
 if [ -z "$ANDROID_NDK" -o -z "$ANDROID_SDK" ]; then
    echo "You must define ANDROID_NDK, ANDROID_SDK and ANDROID_ABI before starting."
    echo "They must point to your NDK and SDK directories.\n"
-- 
1.8.3.4

