From 015eaed83d04572caefca276d2054374031f8a03 Mon Sep 17 00:00:00 2001
From: bbcallen <bbcallen@gmail.com>
Date: Sun, 24 Jun 2012 17:23:34 +0800
Subject: [PATCH] build for armeabi-v7a

---
 compile.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/compile.sh b/compile.sh
index fa8cd27..6834cba 100755
--- a/compile.sh
+++ b/compile.sh
@@ -1,5 +1,8 @@
 #! /bin/sh
 
+# patched for armeabi-v7a
+export ANDROID_ABI=armeabi-v7a
+
 # Read the Android Wiki http://wiki.videolan.org/AndroidCompile
 # Setup all that stuff correctly.
 # Get the latest Android SDK Platform or modify numbers in configure.sh and vlc-android/default.properties.
-- 
1.8.3.4

